Добавляет статью OKLCH (#1)
ninaTorgunakova/web-standards.ru: src/articles/oklch/index.md

      min-width: var(--mobile-width);
    }
  }
</style>

<pre data-lang="css">
<code tabindex="0" class="language-css">
<span class="selector">a:hover&nbsp;</span><span>{</span>
  <span class="property" style="--device-width: 150px; --mobile-width: 110px;">background:&nbsp;</span><div class="preview-with-value"><div class="color-preview without-opacity" style="background-color: rgb(0.21, 50.42, 225.59)"></div><span class="value">oklch(45% 0.26 264);&nbsp;</span><span class="comment">/* голубой */</span></div>
  <span class="property" style="--device-width: 150px; --mobile-width: 110px;">color:&nbsp;</span><div class="preview-with-value"><div class="color-preview without-opacity" style="background-color: #fff"></div><span class="value">oklch(100% 0 0);&nbsp;</span><span class="comment">/* белый */</span></div>
  <span class="property" style="--device-width: 150px; --mobile-width: 110px;">color:&nbsp;</span><div class="preview-with-value"><div class="color-preview" style="background-color: rgba(0, 0, 0, 0.5);"></div><span class="value">oklch(0% 0 0 / 50%);&nbsp;</span><span class="comment">/* чёрный с прозрачностью 50% */</span></div>
<span>}</span>
</code>
</pre>

<figure>

>     <img src="images/oklch-picker.png" loading="lazy" alt="Цветовой миксер OKLCH от Evil Martians отображает пространство OKLCH с ползунками для настройки яркости, насыщенности, альфа-канала и оттенка.">

Reviewer: ```suggestion
    <img src="images/oklch-picker.png" loading="lazy" alt="Цветовой миксер OKLCH Злых марсиан отображает пространство OKLCH с ползунками для настройки яркости, насыщенности, альфа-канала и оттенка.">
```

Я на стороне Пепелсбея в деле агрессивной кириллизации. И у Марсиан русское название всем известно и хорошо звучит.

Author: Поняла, сомневалась как раз насчёт этого места

    <figcaption>Пространство OKLCH в цветовом миксере.</figcaption>
</figure>

<pre data-lang="css">
<code tabindex="0" class="language-css">
<span class="selector">.old&nbsp;</span><span>{</span>
  <span class="property">color:&nbsp;</span><div class="preview-with-value"><div class="color-preview without-opacity" style="background-color: rgb(51, 170, 51);"></div><span class="value">rgb(51, 170, 51);&nbsp;</span></div>
  <span class="property">color:&nbsp;</span><div class="preview-with-value"><div class="color-preview" style="background-color: rgba(51, 170, 51, 0.5);"></div><span class="value">rgba(51, 170, 51, 0.5);&nbsp;</span></div>
<span>}</span>

<span class="selector">.new&nbsp;</span><span>{</span>
  <span class="property">color:&nbsp;</span><div class="preview-with-value"><div class="color-preview without-opacity" style="background-color: rgb(51 170 51);"></div><span class="value">rgb(51 170 51);&nbsp;</span></div>
  <span class="property">color:&nbsp;</span><div class="preview-with-value"><div class="color-preview" style="background-color: rgb(51 170 51 / 50%);"></div><span class="value">rgb(51 170 51 / 50%);&nbsp;</span></div>
<span>}</span>
</code>
</pre>

<figure>

>     <img src="images/p3.png" loading="lazy" alt="Слева — фигура, представляющая собой расширяющийся клин и показывающая, как P3-цвета расширяют цветовое многообразие по сравнению с sRGB-пространством. Справа — левая иконка отображается в sRGB-цветах, а более яркая правая иконка отображается в P3-цветах, подчёркивая, насколько они ярче.">

Reviewer: Нравится подробное описание в `alt` 👍

    <figcaption>Слева — недавно ставший доступным набор зелёных P3-цветов. Справа — сравнение иконок, использующих sRGB и P3-цвета соответственно.</figcaption>
</figure>

<pre data-lang="css">
<code tabindex="0" class="language-css">
<span class="comment">/* В качестве примера в следующих CSS-правилах используются hsl().

>    Не используйте формат hsl() в продакшн-сборке, так как это влечёт проблемы с доступностью. */</span>

Reviewer: ```suggestion
   Не используйте формат hsl() в реальных проектах, так как это влечёт проблемы с доступностью. */</span>
```

`продакшн-сборке` тут прямой перевод. Я предлагаю больше отходить от первоисточника ради того, чтобы текст лучше звучал на русском.

Author: Хм, хорошо, спасибо. Неправильно уловила смысл фразы, подумала про этот совет в плане "в процессе разработки проекта можете использовать, но в конечную версию помещать не следует"

Reviewer: Вообще аргумент — исправил оригинальную статью тоже

---
title: 'OKLCH в CSS: почему мы ушли от RGB и HSL'
date: 2022-21-11
author:
    - andrey-sitnik
    - nina-torgunakova
tags:
    - css
---

<style>
  .color-preview {
    width: 16px;
    height: 16px;
    border: 1px solid black;
    box-sizing: border-box;
    display: inline-block;
    position: relative;
    top: 2px;
    margin-right: 2px;
  }

  .color-preview::before {
    background: repeating-conic-gradient(#fff 0% 25%, #999 0% 50%) 0 0 / 14px 14px;
    display: block;
    width: 14px;
    height: 14px;
    content: '';
    z-index: -1;
    opacity: 0.5;
  }

  .color-preview.without-opacity::before {
    display: none;
  }

  .preview-with-value {
    display: inline-block;
    position: relative;
  }

  .selector {
    color: #a1490f;
  }

  .property {
    --device-width: 80px;
    --mobile-width: 65px;
    color: #550098;
    display: inline-block;
    min-width: var(--device-width);
  }

  .comment {
    color: #706D71;
  }

  @media (max-width: 1239px) {
    .property {
      min-width: var(--mobile-width);
    }
  }
</style>

> В CSS мы чаще всего пишем цвета через`rgb()`или hex — но так сложилось исторически. Новая спецификация CSS Color 4 позволит нам описывать цвета удобнее. В этой статье мы расскажем, почему нам больше всего нравится`oklch()`.

Reviewer: ```suggestion
В CSS мы чаще всего пишем цвета через `rgb()` или hex — но так сложилось исторически. Новая спецификация CSS Color 4 позволит нам описывать цвета удобнее. В этой статье мы расскажем, почему нам больше всего нравится `oklch()`.
```

Пример пропущенных пробелов

Author: А, поняла, про что ты. Я первоначально писала с пробелами, но меня отпугнул такой внешний вид (не нравится, что отступы такие большие становятся):
![screen](https://user-images.githubusercontent.com/78731462/204517748-bc22f4eb-a8af-41e9-b652-244997e47741.jpg)
Но вижу, что в других статьях так же, поэтому отредактирую у себя

## Краткое объяснение

`oklch()`— новый способ определять цвета в CSS. В`oklch(L C H)`или`oklch(L C H / a)` компоненты из аббревиатуры расшифровываются так:

> - `L`— яркость (`0%`-`100%`). Отметим, что она передаётся без искажения, в отличие от`L`в`hsl()`;

Reviewer: Мне больше нравится «как видят глаза» (которые ты используешь ниже) вместо «без искажений». Так не понятно что за искажения.

Author: Поправила

- `C`— насыщенность. Варьируется от серого до наиболее интенсивного оттенка;

> - `H`— оттенок. В качестве значения указывается угол поворота оттенка на цветовом круге (`0`-`360`);

Reviewer: ```suggestion
- `H`— оттенок, угол поворота оттенка на цветовом круге (`0`-`360`);
```
Можно так сократить и убрать сложную для понимания «В качестве значения»

Author: Может после тире поставить просто "_угол поворота оттенка на цветовом круге (`0`-`360`);_"?

Твой вариант хорошо упростит, но меня немного смущает тавтология с формами слова "оттенок"

Reviewer: Сложно получается. Что насчёт

```
оттенок, угол поворота на цветовом круге (`0`-`360`)
```

Author: Думаю, отлично

Добавила

<pre data-lang="css">
<code tabindex="0" class="language-css">
<span class="selector">a:hover&nbsp;</span><span>{</span>
  <span class="property" style="--device-width: 150px; --mobile-width: 110px;">background:&nbsp;</span><div class="preview-with-value"><div class="color-preview without-opacity" style="background-color: rgb(0.21, 50.42, 225.59)"></div><span class="value">oklch(45% 0.26 264);&nbsp;</span><span class="comment">/* голубой */</span></div>
  <span class="property" style="--device-width: 150px; --mobile-width: 110px;">color:&nbsp;</span><div class="preview-with-value"><div class="color-preview without-opacity" style="background-color: #fff"></div><span class="value">oklch(100% 0 0);&nbsp;</span><span class="comment">/* белый */</span></div>
  <span class="property" style="--device-width: 150px; --mobile-width: 110px;">color:&nbsp;</span><div class="preview-with-value"><div class="color-preview" style="background-color: rgba(0, 0, 0, 0.5);"></div><span class="value">oklch(0% 0 0 / 50%);&nbsp;</span><span class="comment">/* чёрный с прозрачностью 50% */</span></div>
<span>}</span>
</code>
</pre>
Формат OKLCH обладает многими преимуществами:

> 1. В отличие от`rgb()`или hex(`#ca0000`), OKLCH легко читается. По числам внутри`oklch()`мы без труда сможем определить записанный цвет. Схоже с удобством формата HSL, но тот не передаёт яркость так, как её видит человек;

Reviewer: ```suggestion
1. В отличие от `rgb()` или hex (`#ca0000`), OKLCH легко читается. По числам внутри `oklch()` мы без труда сможем определить записанный цвет. Схоже с удобством формата HSL, но тот не передаёт яркость так, как её видит человек;
```

Пример отсутствия пробела перед скобкой

> 2. OKLCH лучше подходит для трансформаций цвета, чем`hsl()`. Он правильно передаёт яркость, поэтому изменение оттенка сохранит контраст. Это защищает от неожиданных результатов, которые могут возвращаться CSS-функциями (например,`darken()`в Sass);

Reviewer: ```suggestion
2. OKLCH лучше подходит для измения цвета, чем `hsl()`. Он правильно передаёт яркость, поэтому изменение оттенка сохранит контраст. Это защищает от неожиданных результатов, которые были у `darken()` в Sass;
```

Не уверен о каких CSS-функциях речь, давай короче запишем

Author: Поправила своим коммитом (в саджет опечатка попала)

> 3. OKLCH имеет лучшую доступность благодаря предсказуемой яркости. Это очень заметно при создании палитр для дизайн-систем;

Reviewer: Давай причину (предсказуемый контраст) поставим вперёд, чтобы было сразу понятно.

Наприер:

```suggestion
3. Из-за предсказуемого контраста у OKLCH гораздо лучше с доступностью. Это очень важно при создании палитр для дизайн-систем;
```

> 4. Многие новые устройства (например, от Apple) поддерживают больше цветов, чем старые мониторы sRGB. OKLCH позволяет использовать гамму P3, которая шире стандартного набора. В формате OKLCH действительно удобнее работать с новыми цветами.

Reviewer: ```suggestion
4. Многие новые устройства (например, от Apple) поддерживают больше цветов, чем старые мониторы sRGB. OKLCH позволяет использовать эти P3-цвета.
```

Давай во введении не погружать в сложный термин «гамма». Я ещё убрал вывод, так как не понятно как он связан с P3.

Но, работая с OKLCH, следует помнить о паре важных нюансов:

1. Для OKLCH или LCH при подборе `L`,`C`и`H`есть риск получить цвет, выходящий за пределы возможностей экрана. Хоть браузеры и попытаются найти ближайший поддерживаемый цвет, нам стоит проверять результат в цветовом миксере;

> 2. OKLCH — цветовое пространство, которое появилось совсем недавно. На момент написания этой статьи в 2022 году его экосистема ограничена. Но у нас уже есть`oklch()`<a href="https://github.com/csstools/postcss-plugins/tree/main/plugins/postcss-oklab-function" target="_blank" rel="noopener noreferrer">полифил</a>, а также <a href="https://huetone.ardov.me/" target="_blank" rel="noopener noreferrer">генератор палитры</a>, <a href="https://oklch.evilmartians.io/" target="_blank" rel="noopener noreferrer">цветовой миксер</a> и много <a href="https://bottosson.github.io/posts/oklab/#oklab-implementations" target="_blank" rel="noopener noreferrer">конвертеров</a>.

Reviewer: ```suggestion
2. OKLCH — цветовое пространство, которое появилось совсем недавно. На момент написания этой статьи в 2022 году его экосистема только растёт. Но у нас уже есть <a href="https://github.com/csstools/postcss-plugins/tree/main/plugins/postcss-oklab-function" target="_blank" rel="noopener noreferrer">полифил</a>, <a href="https://huetone.ardov.me/" target="_blank" rel="noopener noreferrer">генератор палитры</a>, <a href="https://oklch.evilmartians.io/" target="_blank" rel="noopener noreferrer">цветовой миксер</a> и много <a href="https://bottosson.github.io/posts/oklab/#oklab-implementations" target="_blank" rel="noopener noreferrer">конвертеров</a>.
```

Немного сократил и нежнее написал про проблему :)

Reviewer: А не хочешь ссылки писать как:

```
[полфил](https://github.com/csstools/postcss-plugins/tree/main/plugins/postcss-oklab-function)
```

Будет читаемее и пусть сам сайт решает где открывать ссылки в новом окне

Author: Спасибо, очень полезно!

<figure>
    <img src="images/oklch-picker.png" loading="lazy" alt="Цветовой миксер OKLCH Злых марсиан отображает пространство OKLCH с ползунками для настройки яркости, насыщенности, альфа-канала и оттенка.">
    <figcaption>Пространство OKLCH в цветовом миксере.</figcaption>
</figure>

> Это было лишь краткое описание особенностей OKLCH. Если вы готовы больше узнать о новой эре CSS-цветов, то смело переходите к первой главе.

Reviewer: Ощущение, что это стиль статей на английском. Не по языку, а по стилю.

Может убрать? Или на что-то совсем другое заменить?

Author: Имеешь в виду, что закидывание удочки на чтение нового раздела не очень свойственно русским статьям? Или что-то иное?

Reviewer: Ага, давай попробуем убрать

Author: Убрала, интересная тема для размышлений, хочу потом ещё её поисследовать

<pre data-lang="css">
<code tabindex="0" class="language-css">
<span class="selector">.old&nbsp;</span><span>{</span>
  <span class="property">color:&nbsp;</span><div class="preview-with-value"><div class="color-preview without-opacity" style="background-color: rgb(51, 170, 51);"></div><span class="value">rgb(51, 170, 51);&nbsp;</span></div>
  <span class="property">color:&nbsp;</span><div class="preview-with-value"><div class="color-preview" style="background-color: rgba(51, 170, 51, 0.5);"></div><span class="value">rgba(51, 170, 51, 0.5);&nbsp;</span></div>
<span>}</span>

<span class="selector">.new&nbsp;</span><span>{</span>
  <span class="property">color:&nbsp;</span><div class="preview-with-value"><div class="color-preview without-opacity" style="background-color: rgb(51 170 51);"></div><span class="value">rgb(51 170 51);&nbsp;</span></div>
  <span class="property">color:&nbsp;</span><div class="preview-with-value"><div class="color-preview" style="background-color: rgb(51 170 51 / 50%);"></div><span class="value">rgb(51 170 51 / 50%);&nbsp;</span></div>
<span>}</span>
</code>
</pre>

> Но что ещё важнее, CSS Color 4 добавляет 14 новых способов для определения цвета. И это не просто синтаксический сахар. Новые форматы записи (среди которых и`oklch()`) улучшают читаемость кода, доступность и позволяют расширить возможности наших сайтов.

Reviewer: ```suggestion
Но что ещё важнее, CSS Color 4 добавляет 14 новых способов для определения цвета. И это не просто синтаксический сахар. Новые форматы записи (среди которых и `oklch()`) улучшают читаемость кода, доступность и могут принести прямую пользу для наших сайтов.
```

Кажется так будет более по-русски.

#### P3 Colors

> Стандартный набор цветов, который поддерживают устройства, называется sRGB. В него входят не все цвета, которые способен различить человеческий глаз, — это выходит за грани возможностей многих мониторов.

Reviewer: > Стандартный набор цветов, который поддерживают устройства, называется sRGB.

Сложно получилось.

Может сначала расскажем что мониторы поддерживают не все цвета глаза. Потом скажем что текущий набор цветов называется «sRGB»?

Author: Изменила формулировку на "_Человеческий глаз способен различить гораздо больше цветов, чем способны отобразить многие устройства. Текущий, стандартно поддерживаемый набор цветов называется sRGB._"

Стало ли лучше?

Reviewer: > Текущий, стандартно поддерживаемый набор цветов

Вот тут самое сложное. Непонятно к чему «стандартно» (к набору или поддерживаемому).

Лучше явно разбить «паровозик» на группы: «Текущий стандарт набора цветов называется sRGB».

Author: Отлично, чётко звучит. Изменила

> Но уже сейчас все современные устройства Aple и многие OLED-экраны поддерживают на 30% больше цветов, чем есть в sRGB. Этот расширенный набор цветов называется P3. Он также известен, как широкий цветовой охват.

Reviewer: ```suggestion
Но уже сейчас все современные устройства Apple и многие OLED-экраны поддерживают на 30% больше цветов, чем есть в sRGB. Этот расширенный набор цветов называется P3. Он также известен, как широкий цветовой охват (wide-gamut colors).
```

Давай продублируем английский термин.

Поддержка дополнительных 30% цветов будет полезна для дизайнеров:

> - Новые цвета зачастую заметно ярче старых. Это поможет создавать более привлекательные и красочные веб-страницы;

Reviewer: ```suggestion
- Новые цвета зачастую заметно насыщеннее старых. Это поможет создавать более привлекательные и красочные веб-страницы;
```

Это именно больше Chroma а не Lightness.

> - Дополнительные цвета делают более гибким процесс создания палитр для дизайн-систем.

Reviewer: ```suggestion
- Больше цветов делают более гибким процесс создания палитр для дизайн-систем.
```

Упростим

<figure>

>     <img src="images/p3.png" loading="lazy" alt="Слева — фигура, представляющая собой расширяющийся клин и показывающая, как P3-цвета расширяют цветовое многообразие по сравнению с sRGB-пространством. Справа — левая иконка отображается в sRGB-цветах, а более яркая правая иконка отображается в P3-цветах, подчёркивая, насколько они ярче.">

Reviewer: Напоминаю отправить PR в https://github.com/evilmartians/evilmartians.com/blob/master/content/posts/2022-10-25-oklch-in-css-why-quit-rgb-hsl/index.md

Author: Создала PR https://github.com/evilmartians/evilmartians.com/pull/718

    <figcaption>Слева — недавно ставший доступным набор зелёных P3-цветов. Справа — сравнение иконок, использующих sRGB и P3-цвета соответственно.</figcaption>
</figure>

Итак, теперь у нас появились P3-цвета — но радоваться ещё рано: чтобы их использовать, нам нужно найти формат, который поддерживает охват P3.`rgb()`,`hsl()`или hex неудобны для указания P3-цветов — хоть мы и можем использовать новую запись`color(display-p3 1 0 0)`, это не решает проблему плохой читаемости RGB.

К нашему везению, OKLCH очень легко читать и менять прямо в коде. К тому же он поддерживает не только P3, но и любые цвета, которые способен увидеть человек.

> #### Нативная поддержка работы с цветом в CSS

Reviewer: ```suggestion
#### Изменение цвета прямо в CSS
```

Давай избежим «нативный» и вместо «работа» (абстрактное понятие) напишем что-то конкретное

Author: Хм, может ещё подойдёт "Изменение цвета в CSS напрямую" или, например, "Встроенные изменения цветов в CSS"?

Reviewer: «Встроенные изменения цветов в CSS» нравится

Author: Добавила

> CSS Color 4 — это большой шаг в мире стилей, и грядущий <a href="https://www.w3.org/TR/css-color-5/" target="_blank" rel="noopener noreferrer">CSS Color 5</a> будет ещё полезнее. В нём, наконец, появится нативная поддержка работы с цветом.

Reviewer: Лучше продублировать тут ссылку на спеку

Reviewer: `и грядущий` → `но грядущий`

Author: Добавила спеку, поправила предлог

<span class="selector">:root&nbsp;</span><span>{</span>
  <span class="property">--accent:&nbsp;</span><div class="preview-with-value"><div class="color-preview without-opacity" style="background-color: hsl(63 61% 40%);"></div><span class="value">hsl(63 61% 40%);&nbsp;</span></div>
<span>}</span>

<span class="selector">.error&nbsp;</span><span>{</span>
  <span class="comment">/* Красный вариант акцентирующего цвета */</span>
  <span class="property">background:&nbsp;</span><div class="preview-with-value"><div class="color-preview without-opacity" style="background-color: hsl(20, 61%, 40%);"></div><span class="value">hsl(from var(--accent) 20 s l);&nbsp;</span></div>
<span>}</span>

<span class="selector">.button:hover&nbsp;</span><span>{</span>
  <span class="comment">/* Вариант на 10% светлее */</span>
  <span class="property">background:&nbsp;</span><div class="preview-with-value"><div class="color-preview without-opacity" style="background-color: hsl(63, 61%, 50%);"></div><span class="value">hsl(from var(--accent) h s calc(l + 10%));&nbsp;</span></div>
<span>}</span>
</code>
</pre>

> С новым синтаксисом мы можем взять один цвет — например, из кастомного свойства — и легко изменить его, переписав отдельные компоненты.

Reviewer: ```suggestion
С новым синтаксисом мы можем взять один цвет — например, из кастомного свойства — и легко изменить его, переписав отдельные компоненты цвета.
```

Чуть понятнее

> Но, как мы упоминали, в подходе с использованием`hsl()`есть ощутимый недостаток: этот формат обладает плохой доступностью. Для него значение`l`зависит от угла поворота оттенка, поэтому цвет при использовании`hsl()`может удивить неожиданным контрастом.

Reviewer: ```suggestion
Но, как мы упоминали, у `hsl()` есть ощутимый недостаток: этот формат обладает плохой доступностью. Для него значение `l` зависит от угла поворота оттенка, поэтому после изменения через `hsl()`, может получится плохой контраст текста и фона.
```

Чуть подробнее

Author: Да, стало лучше. Применила этот саджест своим коммитом (чуть поправив орфографию)

<pre data-lang="css">
<code tabindex="0" class="language-css">
<span class="selector">:root&nbsp;</span><span>{</span>
  <span class="property">--accent:&nbsp;</span><div class="preview-with-value"><div class="color-preview without-opacity" style="background-color: rgb(160, 167, 45);"></div><span class="value">oklch(70% 0.14 113);&nbsp;</span></div>
<span>}</span>

<span class="selector">.error&nbsp;</span><span>{</span>
  <span class="comment">/* Красный вариант акцентирующего цвета */</span>
  <span class="property">background:&nbsp;</span><div class="preview-with-value"><div class="color-preview without-opacity" style="background-color: rgb(232, 119, 130);"></div><span class="value">oklch(from var(--accent) l c 15);&nbsp;</span></div>
<span>}</span>

<span class="selector">.button:hover&nbsp;</span><span>{</span>
  <span class="comment">/* Вариант на 10% светлее */</span>
  <span class="property">background:&nbsp;</span><div class="preview-with-value"><div class="color-preview without-opacity" style="background-color: rgb(190, 199, 82);"></div><span class="value">oklch(from var(--accent) calc(l + 10%) c h);&nbsp;</span></div>
<span>}</span>
</code>
</pre>

> Отметим: вам не нужно использовать OKLCH для указания`--accent`в`oklch(from …)`, но для улучшения читаемости мы рекомендуем писать в едином формате.

Reviewer: ```suggestion
Отметим: `--accent` в `oklch(from …)` не обязательно должен быть тоже в формате OKLCH. Но для улучшения читаемости мы рекомендуем писать в едином формате.
```

date: 2022-21-11
author:
    - andrey-sitnik
    - nina-torgunakova
tags:
    - css
---

<style>
  .text-highlight {
    background-color: #0000000f;
    border-radius: 4px;
    padding: 0.2em 6px;
    white-space: pre-wrap;
    word-break: break-word;
  }

  .color-preview {
    width: 16px;
    height: 16px;
    border: 1px solid black;
    box-sizing: border-box;
    display: inline-block;
    position: relative;
    top: 2px;
    margin-right: 2px;
  }

  .color-preview::before {
    background: repeating-conic-gradient(#fff 0% 25%, #999 0% 50%) 0 0 / 14px 14px;
    display: block;
    width: 14px;
    height: 14px;
    content: '';
    z-index: -1;
    opacity: 0.5;
  }

  .color-preview.without-opacity::before {
    display: none;
  }

  .preview-with-value {
    display: inline-block;
    position: relative;
  }

  .selector {
    color: #a1490f;
  }

  .property {
    --device-width: 80px;
    --mobile-width: 65px;
    color: #550098;
    display: inline-block;
    min-width: var(--device-width);
  }

  .comment {
    color: #706D71;
  }

  @media (max-width: 1239px) {
    .property {
      min-width: var(--mobile-width);
    }
  }
</style>

> При работе с CSS мы чаще всего указываем цвета с помощью <span class="text-highlight">rgb()</span> или hex – так сложилось исторически. Однако новая спецификация CSS Color 4 открывает для разработчиков более перспективные способы объявления цветов. В этой статье мы расскажем, почему из всех возможных вариантов мы выбрали именно <span class="text-highlight">oklch()</span>.

Reviewer: ```suggestion
При работе с CSS мы чаще всего указываем цвета с помощью `rgb()` или hex – так сложилось исторически. Однако новая спецификация CSS Color 4 открывает для разработчиков более перспективные способы объявления цветов. В этой статье мы расскажем, почему из всех возможных вариантов мы выбрали именно `oklch()`.
```

Мне кажется тут должен быть `<code>` раз это пример кода. На сайте Веб-стандартов должен же быть хороший стиль для них?

Reviewer: `−` это короткое тире, оно используется в английском.

В русском используется длинное тире `—`.

Кстати, в Линуксе можно включить «[типографическую раскладку](http://s.arboreus.com/2010/09/typography-keyboard-layout-in-linux.html)», которая похожа на [раскладку Бирмана](https://ilyabirman.ru/typography-layout/).

С неё ввод `—` будет <kbd>AltGr</kbd>+<kbd>-</kbd>.

Reviewer: По стилю предлагаю два изменения:

1. Написать проще и короче. Твой стиль тоже норм, но мне кажется тут сработает больше формат «вы с другом собрались в баре и он рассказывает про новую технологию».
2. Предлагаю активнее использовать союзы для управления эмоциями читателя. Вот тут «Однако» не создаёт положительное настроение.

Понимаю, что предложения немного абстрактные, поэтому вот пример:

> В CSS мы чаще всего пишем цвета через `rgb()` или hex — но так сложилось исторически. Новая спецификация CSS Color 4 позволит нам описывать цвета удобнее. В этой статье мы расскажем, почему нам больше всего нравится `oklch()`.

Что тут изменилась:
— Длинные конструкции убрал «При работе с CSS» → «В CSS»
— Добавил больше эмоций «мы выбрали» → «больше всего нравится»
— Ближе к народу «открывает для разработчиков» → «позволит нам».

Reviewer: И нужно не забыть ссылку на спеку https://www.w3.org/TR/css-color-4/

Author: Да, на сайте Веб-стандартов дефолтный стиль для кода выглядит так:
![screen](https://user-images.githubusercontent.com/78731462/204133396-4d40467a-1649-4bda-a60e-db9404d512a1.jpg)

## Краткое объяснение

> <span class="text-highlight">oklch()</span> – это новый способ определять CSS-цвета. В <span class="text-highlight">oklch(L C H)</span> или <span class="text-highlight">oklch(L C H / a)</span> компоненты, представленные буквами в аббревиатуре, означают следующее:

Reviewer: ```suggestion
<span class="text-highlight">oklch()</span> — новый способ определять цвета в CSS. В <span class="text-highlight">oklch(L C H)</span> или <span class="text-highlight">oklch(L C H / a)</span> компоненты, представленные буквами в аббревиатуре, означают следующее:
```

<span>}</span>

<span class="selector">.error&nbsp;</span><span>{</span>
  <span class="comment">/* Красный вариант акцентирующего цвета */</span>
  <span class="property">background:&nbsp;</span><div class="preview-with-value"><div class="color-preview without-opacity" style="background-color: rgb(232, 119, 130);"></div><span class="value">oklch(from var(--accent) l c 15);&nbsp;</span></div>
<span>}</span>

<span class="selector">.button:hover&nbsp;</span><span>{</span>
  <span class="comment">/* Вариант на 10% светлее */</span>
  <span class="property">background:&nbsp;</span><div class="preview-with-value"><div class="color-preview without-opacity" style="background-color: rgb(190, 199, 82);"></div><span class="value">oklch(from var(--accent) calc(l + 10%) c h);&nbsp;</span></div>
<span>}</span>
</code>
</pre>
Отметим: `--accent` в `oklch(from …)` не обязательно должен быть тоже в формате OKLCH. Но для улучшения читаемости мы рекомендуем писать в едином формате.

## Сравнение OKLCH с другими CSS-форматами цветов

> В предыдущей главе мы кратко рассказали об `oklch()`, P3 и встроенных изменениях цвета. Надеемся, что теперь вы узнали больше о развитии цвета в CSS.

Reviewer: Может стоит эту фразу тоже убрать? Как-то она пользы несёт и в английском тексте выглядит чть лучше.

> Мы можем комбинировать разные форматы при работе с цветами. Необязательно везде использовать OKLCH; но единый подход к записи значительно улучшает читаемость и поддерживаемость кода.

Reviewer: ```suggestion
Мы можем комбинировать разные форматы при работе с цветами. Необязательно везде использовать `oklch()`; но единый подход к записи значительно улучшает читаемость и поддерживаемость кода.
```

Тут лучше будет про функцию сразу написать (может стоит исправить англ. версию тоже?)

Author: Ага, исправила, создам PR с правкой в оригинальной

Author: https://github.com/evilmartians/evilmartians.com/pull/721

> А что, если мы постараемся выбрать единый, универсальный формат записи цвета? Попробуем развить эту идею и определить критерии для выбора:

Reviewer: ```suggestion
Давайте постараемся выбрать единый, универсальный формат записи цвета? Наши критерии для выбора:
```

Чуть короче

Author: Мне нравится, только, может, `Давайте попробуем` вместо `Давайте постараемся`? Мне кажется, прозвучит более ненавязчиво.

Reviewer: Ага, хорошая идея

> 1. Для него должна быть встроенная поддержка в CSS;

Reviewer: ```suggestion
1. У него должна быть встроенная поддержка в CSS;
```

2. Он должен уметь работать как минимум с широким цветовым охватом P3;

> 3. Он подойдёт для лёгкого изменения цвета. Для этого компоненты формата должны быть удобочитаемыми и при этом не зависеть друг от друга. Например, изменение яркости должно сохранять прежний уровень контраста, а изменение насыщенности не должно менять оттенок.

Reviewer: ```suggestion
3. Он должен хорошо подходить для изменения цвета. Компоненты формата должны быть удобочитаемыми и при этом не зависеть друг от друга. Например, изменение яркости должно сохранять прежний уровень контраста, а изменение насыщенности не должно менять оттенок.
```

> #### OKLCH против RGB/Hex

Reviewer: ```suggestion
#### OKLCH против RGB и hex
```

> Все форматы `rgb(109 162 218)`, `#6ea3db` или аналог P3-цветов `color(display-p3 0.48 0.63 0.84)` записываются тремя числами, которые определяют количество красного, зелёного и голубого соответственно. Обратите внимание: `1` в`color(display-p3)` шифрует большее значение, чем `255` в RGB.

Reviewer: ```suggestion
Все форматы `rgb(109 162 218)`, `#6ea3db` или P3-аналог `color(display-p3 0.48 0.63 0.84)` записываются тремя числами, которые определяют количество красного, зелёного и голубого соответственно. Обратите внимание: `1` в`color(display-p3)` кодирует большее значение, чем `255` в RGB.
```

> Какая проблема объединяет все эти форматы? Они абсолютно нечитаемы для большинства разработчиков. Люди часто начинают обращаться с их составляющими, как с магическими числами, не пытаясь вникнуть в их смысл.

Reviewer: ```suggestion
Какая проблема объединяет все эти форматы? Они абсолютно нечитаемы для большинства разработчиков. Люди часто просто копируют их как магический набор цифр, не пытаячь их прочитать и понять.
```

RGB, hex и `color(display-p3)` неудобны для трансформаций цвета, поскольку большинству людей интуитивно сложно задавать цвета, изменяя количество красного, голубого и зелёного. Кроме того, с помощью RGB и hex нельзя определить P3-цвета.

> С другой стороны, с помощью OKLCH, LCH и HSL можно установить цвет более похожим на тот, который увидит человек.

Reviewer: ```suggestion
С другой стороны, OKLCH, LCH и HSL кодирует цвет так, как люди о думают о цветах — через оттенок, насыщенность, яркость.
```

<span>}</span>
</code>
</pre>

<pre data-lang="css">
<code tabindex="0" class="language-css">
<span class="selector">.button&nbsp;</span><span>{</span>
  <span class="comment">/* Голубой */</span>
  <span class="property">background:&nbsp;</span><div class="preview-with-value"><div class="color-preview without-opacity" style="background-color: #6ea3db;"></div><span class="value">oklch(70% 0.1 250);</span></div>
<span>}</span>
<span class="selector">.button:hover&nbsp;</span><span>{</span>
  <span class="comment">/* Более яркий голубой */</span>
  <span class="property">background:&nbsp;</span><div class="preview-with-value"><div class="color-preview without-opacity" style="background-color: #7db3eb;"></div><span class="value">oklch(75% 0.1 250);</span></div>
<span>}</span>
<span class="selector">.button.is-delete&nbsp;</span><span>{</span>
  <span class="comment">/* Красный с той же насыщенностью */</span>
  <span class="property">background:&nbsp;</span><div class="preview-with-value"><div class="color-preview without-opacity" style="background-color: #d68585;"></div><span class="value">oklch(70% 0.1 20);</span></div>
<span>}</span>
</code>
</pre>
</div>

> Простой и понятный порядок определения цветов в OKLCH — это очень удобно. Но здесь стоит вспомнить и про обратную сторону монеты, слабейшее место OKLCH: несмотря на все свои преимущества, он появился совсем недавно. Его система ещё находится в стадии активного развития.

Reviewer: ```suggestion
OKLCH-цвета можн прочитать и сравнить. Но у OKLCH появился совсем недавно — его экосистема только находится в стадии активного развития.
```

#### OKLCH против HSL

Теперь сравним OKLCH и HSL. HSL использует 3 компонента для кодировки оттенка, насыщенности и яркости, например `hsl(210 60% 64%)`. Главная проблема с HSL в том, что это цветовое пространство растягивается в цилиндр.

> Каждый оттенок должен иметь одинаковое значение насыщенности (`0%`—`100%`). Но в реальности экран отображает цвет не так, как это видит человеческий глаз: разные оттенки будут обладать неодинаковой максимальной насыщенностью. HSL не обращает внимания на это различие. Он деформирует цветовое пространство и растягивает цвета до одинакового показателя максимальной насыщенности.

Reviewer: ```suggestion
Каждый оттенок должен иметь одинаковое значение насыщенности (`0%`—`100%`). Но в реальности экран и глаз имеют разную максимальную насыщенность для разных оттенков. HSL не обращает внимания на это различие. Он деформирует цветовое пространство и растягивает цвета до одинакового показателя максимальной насыщенности.
```

Там дело не в разности глаза и экрана, реально разные насыщенности у цветов

<figure>
    <img src="images/hsl-vs-oklch.avif" loading="lazy" alt="4 изображения. Два сверху показывают пространства HSL и OKLCH с одинаковыми значениями насыщенности/интенсивности. Два снизу — аналогично для черного и белого цвета.">
    <figcaption>Так выглядят одинаковые значения яркости и оттенка в пространствах HSL и OKLCH для цветной и черно-белой версии. Разные оттенки в HSL имеют разную яркость.</figcaption>
</figure>

> HSL искажает итоговый результат, поэтому его нельзя полноценно использовать для работы с цветом. Компонент `L` в нём является непредсказуемым, зависящим от угла поворота оттенка. Это приводит к неприятным проблемам с контрастом и доступностью.

Reviewer: ```suggestion
HSL искажает итоговый результат, поэтому через него нельзя нельзя менять цвет. Компонент `L` в нём является непредсказуемым, зависящим от оттенка. Это приводит к неприятным проблемам с контрастом и доступностью.
```

Вот несколько реальных примеров, где проявляются эти проблемы:

1. Добавление 10% яркости для голубого или для фиолетового цвета приведут к разным результатам. Если вы когда-нибудь использовали функцию `darken()` в SASS, то понимаете, какой непредсказуемый может быть итог;
2. Если вы поменяете оттенок (например, чтобы из фирменного цвета сделать красный цвет ошибки), то с ним может измениться яркость, и текст на этом фоне станет нечитаемым.

<figure>
    <img src="images/buttons-hsl-vs-oklch.avif" loading="lazy" alt="4 кнопки. В первой колонке представлено пространство HSL, во второй — OKLCH. Показано, что при изменении угла поворота оттенка в HSL контраст между текстом и фоном кнопки заметно ухудшается. Для OKLCH такой проблемы нет.">
    <figcaption>Изменение угла в пространстве HSL может привести к проблемам доступности из-за низкого контраста.</figcaption>
</figure>

> HSL плох для работы с цветом. В нашем сообществе многие [просят избегать](https://stripe.com/blog/accessible-color-systems) HSL при создании палитр. Кроме того, подобно RGB или hex, формат HSL не может быть использован для определения P3-цветов.

Reviewer: ```suggestion
HSL плох для работы с цветом. В сообществе многие [просят избегать](https://stripe.com/blog/accessible-color-systems) HSL при создании палитр. Кроме того, подобно RGB или hex, формат HSL не может быть использован для определения P3-цветов.
```

Чтобы не читалось как в «сообществе Марсиан»

> OKLCH не искажает цветовое пространство; этот формат показывает нам реальные цвета со всей красочностью. Это помогает добиться предсказуемого контраста при изменениях цвета и при указании P3-цветов.

Reviewer: ```suggestion
OKLCH не искажает цветовое пространство; этот формат показывает нам реальную физику цвета. Это помогает добиться предсказуемого контраста при изменениях цвета и при указании P3-цветов.
```

Цвета-то одинаковые

С другой стороны, некоторые комбинации значений для OKLCH порождают цвета, которые обычные экраны отобразить не способны. Ряд из них можно увидеть только на P3-мониторах. Но это не критическая проблема: браузеры будут искать ближайший поддерживаемый цвет.

#### OKLCH против Oklab & LCH против Lab

В CSS есть две функции для пространства Oklab: `oklab()` и `oklch()`. Аналогичные есть и для Lab: `lab()` и `lch()`. Так в чём же разница?

> При использовании одинакового пространства есть разные способы задать в нём точку. Oklab и lab существуют в декартовых координатах с осями `a` и `b` (`a`— количество красного/зеленого для цвета, `b` — голубого/желтого). OKLCH и LCH используют полярные координаты, где есть угол оттенка и расстояние для насыщенности.

Reviewer: ```suggestion
При использовании одинакового пространства есть разные способы задать в нём точку. Oklab и lab существуют в декартовых координатах с осями `a` и `b` (`a`— от красного до зелёного, `b` — от голубого до желтого). OKLCH и LCH используют полярные координаты, где есть угол оттенка и расстояние для насыщенности.
```

`a` это именно выбор между красным или синими

Reviewer: Кстати, проверь что везде, где звук «ё» стоит буква `ё`. Обязательная ёфикация — политика редакции (моя и Макеева).

Author: Проверила, конкретно в этом месте опечаталась.

<figure>
    <img src="images/oklab-vs-oklch.avif" loading="lazy" alt="2 круга. Первый иллюстрирует декартовы координаты `a`, `b` в Oklab и прямой угол между этими осями. Второй отображает полярные координаты в OKLCH в виде острого угла поворота оттенка. Угол представлен двумя лучами, один из которых представляет собой насыщенность.">
    <figcaption>Декартовы координаты для Oklab и полярные координаты для OKLCH.</figcaption>
</figure>

> Вкратце, и OKLCH, и LCH являются отличным вариантом для создания читаемого кода и работы с цветом. Значения насыщенности и оттенков для них более приближены к тому, как люди их воспринимают.

Reviewer: ```suggestion
OKLCH, и LCH удобнее для работы с цветом, так как люди думают о цвете через насыщенность и оттенок, а не количество красного и зелённого.
```

Смысл чуть другой

Author: Очень понравилась предложенная формулировка

#### OKLCH против LCH 

> Формат [LCH](https://lea.verou.me/2020/04/lch-colors-in-css-what-why-and-how/) построен поверх пространства CIE LAB (Lab). Он решает все проблемы, которые существуют в HSL и RGB. Также он позволяет работать с P3-цветами, и в большинстве случаев изменение цвета даёт предсказуемый результат.

Reviewer: ```suggestion
Формат [LCH](https://lea.verou.me/2020/04/lch-colors-in-css-what-why-and-how/) построен на пространстве CIE LAB (Lab). Он решает все проблемы, которые существуют в HSL и RGB. Также он позволяет работать с P3-цветами, и в большинстве случаев изменение цвета даёт предсказуемый результат.
```

«Поверх» подразумевает больше изменений

> Однако формат LCH имеет одну неприятную проблему: неожиданный сдвиг оттенка при изменении насыщенности и яркости для голубого цвета (угол поворота между `270` и `330`).

Reviewer: ```suggestion
Однако формат LCH имеет одну неприятную проблему: неожиданный сдвиг оттенка при изменении насыщенности и яркости для голубого цвета (оттенок между `270` и `330`).
```

Author: Может, стоит дописать в конце в скобках `градусами`?

Reviewer: Да, хорошая идея

<figure>
    <img src="images/lch-vs-oklch.avif" loading="lazy" alt="Два треугольника, которые представляют собой срезы пространств LCH и OKLCH, где яркость и насыщенность изменяются, а оттенок — одинаков. Фигура слева, обозначающая LCH, является голубой с одной стороны, и фиолетовой с другой. Фигура справа, обозначающая OKLCH, имеет одинаковый цвет на всей площади, как и ожидается.">
    <figcaption>Срезы пространств LCH и OKLCH, где яркость и насыщенность изменяются, а оттенок — одинаков. Срез LCH голубой с одной стороны и фиолетовый с другой. Оттенок в OKLCH остаётся постоянным, как и ожидается.</figcaption>
</figure>

> Небольшой пример из реального проекта:

Reviewer: ```suggestion
Небольшой пример из реальной жизни:
```

Такого проекта нет :D

Author: Ок, автоматически подумала при чтении, что есть жизнь разработчика, если не проекты :)

<code tabindex="0" class="language-css">
<span class="selector">.temperature.is-very-very-cold&nbsp;</span><span>{</span>
  <span class="property">background:&nbsp;</span><div class="preview-with-value"><div class="color-preview without-opacity" style="background-color: rgb(64, 43, 241)"></div><span class="value">oklch(48% 0.27 274);</span></div>
  <span class="comment">/* Выглядит голубым */</span>
<span>}</span>
<span class="selector">.temperature.is-very-cold&nbsp;</span><span>{</span>
  <span class="property">background:&nbsp;</span><div class="preview-with-value"><div class="color-preview without-opacity" style="background-color: rgb(66, 75, 195);"></div><span class="value">oklch(48% 0.185 274);</span></div>
  <span class="comment">/* Всё ещё голубой */</span>
<span>}</span>
<span class="selector">.temperature.is-cold&nbsp;</span><span>{</span>
  <span class="property">background:&nbsp;</span><div class="preview-with-value"><div class="color-preview without-opacity" style="background-color: rgb(76, 88, 150);"></div><span class="value">oklch(48% 0.1 274);</span></div>
  <span class="comment">/* Всё ещё голубой */</span>
<span>}</span>
</code>
</pre>
</div>

> Пространства Oklab и OKLCH [были созданы](https://bottosson.github.io/posts/oklab/) для решения бага со сдвигом оттенка.

Reviewer: ```suggestion
Пространства Oklab и OKLCH [были созданы](https://bottosson.github.io/posts/oklab/) для решения этой проблемы со сдвигом оттенка.
```

Помню, что хочется чуть разговорнее, но хочется по возможности избежать заимствований

> Но OKLCH — не просто исправление. В нём появляется [много полезных возможностей](https://www.w3.org/TR/css-color-4/#ok-lab), связанных с математическими преобразованиями. Например, появляется улучшенная [гамма-коррекция](https://bottosson.github.io/posts/gamutclipping/), и CSSWG [рекомендует использовать OKLCH](https://www.w3.org/TR/css-color-4/#css-gamut-mapping) для гамма-отображения.

Reviewer: ```suggestion
Но OKLCH — не просто исправление. В нём появляется [много полезных возможностей](https://www.w3.org/TR/css-color-4/#ok-lab), связанных с математическими преобразованиями. Например, появляется упрощается [гамма-коррекция](https://bottosson.github.io/posts/gamutclipping/), и CSSWG [рекомендует использовать OKLCH](https://www.w3.org/TR/css-color-4/#css-gamut-mapping) для гамма-коррекция.
```

mapping и correction, вроде, означают одно и то же (или нет?). Давай переводить всегда как коррекцию.

Author: У меня было предположение, что в оригинальной статье под "gamut mapping" имелось в виду что-то вроде "mapping function for gamma correction", т.е. инструмента (функции), который используется для коррекции.

Хорошо.

  <span class="comment">/* Всё ещё голубой */</span>
<span>}</span>
<span class="selector">.temperature.is-cold&nbsp;</span><span>{</span>
  <span class="property">background:&nbsp;</span><div class="preview-with-value"><div class="color-preview without-opacity" style="background-color: rgb(76, 88, 150);"></div><span class="value">oklch(48% 0.1 274);</span></div>
  <span class="comment">/* Всё ещё голубой */</span>
<span>}</span>
</code>
</pre>
</div>

Пространства Oklab и OKLCH [были созданы](https://bottosson.github.io/posts/oklab/) для решения этой проблемы со сдвигом оттенка.
Но OKLCH — не просто исправление. В нём появляется [много полезных возможностей](https://www.w3.org/TR/css-color-4/#ok-lab), связанных с математическими преобразованиями. Например, появляется упрощённая [гамма-коррекция](https://bottosson.github.io/posts/gamutclipping/), и CSSWG [рекомендует использовать OKLCH](https://www.w3.org/TR/css-color-4/#css-gamut-mapping) для гамма-коррекции.

## Принципы работы OKLCH

#### Создание и развитие OKLCH

> [Бьёрн Оттоссон](https://twitter.com/bjornornorn) создал пространства Oklab и OKLCH в 2020 году. Изначально они задумывались для исправления проблемы CIE LAB и LCH. Бьёрн написал [отличную статью](https://bottosson.github.io/posts/oklab/) с описанием деталей реализации. Также в ней он рассказал о причинах, побудивших его разработать эти пространства.

Reviewer: Я попровал озвучить фамилию в Гугл Переводчике и там звучит «утоссон».

[Вики](https://ru.wikipedia.org/wiki/%D0%A8%D0%B2%D0%B5%D0%B4%D1%81%D0%BA%D0%BE-%D1%80%D1%83%D1%81%D1%81%D0%BA%D0%B0%D1%8F_%D0%BF%D1%80%D0%B0%D0%BA%D1%82%D0%B8%D1%87%D0%B5%D1%81%D0%BA%D0%B0%D1%8F_%D1%82%D1%80%D0%B0%D0%BD%D1%81%D0%BA%D1%80%D0%B8%D0%BF%D1%86%D0%B8%D1%8F) говорит что `o` может транслитирироваться как `у`.

Author: Хм, перевела так, основываясь на [странице в Вики](https://ru.wikipedia.org/wiki/%D0%9E%D1%82%D1%82%D0%BE%D1%81%D1%81%D0%BE%D0%BD). У людей из списка "Ottosson" в оригинале. С "Утоссон" или "Уттоссон" не могу ничего найти.

`при сомнении — о` в этой статье означает, что поставить "о" не будет ошибкой?

Reviewer: Хороший способ, да, давай через «О»

> [Бьёрн Оттоссон](https://twitter.com/bjornornorn) создал пространства Oklab и OKLCH в 2020 году. Изначально они задумывались для исправления проблемы CIE LAB и LCH. Бьёрн написал [отличную статью](https://bottosson.github.io/posts/oklab/) с описанием деталей реализации. Также в ней он рассказал о причинах, побудивших его разработать эти пространства.

Reviewer: ```suggestion
[Бьёрн Оттоссон](https://twitter.com/bjornornorn) создал пространства Oklab и OKLCH в 2020 году. Главная задача была в исправлении проблемы CIE LAB и LCH. Бьёрн написал [отличную статью](https://bottosson.github.io/posts/oklab/) с описанием деталей реализации. Также в ней он рассказал о причинах, побудивших его разработать эти пространства.
```

«Изначально» подразумевает, что она изменилась, а тут не так

Стоит отметить, что Oklab появилось гораздо позже других пространств. Изначально это было его главным слабым местом. Но всего два года спустя стало понятно, что сообщество отлично приняло Oklab:

- Оно было добавлено в [спецификацию CSS](https://www.w3.org/TR/css-color-4/#ok-lab);

> - Начиная с версии 15.4., Safari поддерживает `oklch()` и `oklab()` ;

Reviewer: ```suggestion
- Начиная с версии 15.4, Safari поддерживает `oklch()` и `oklab()` ;
```

> - В Photoshop [был добавлен](https://helpx.adobe.com/photoshop/using/gradient-interpolation.html) Oklab для градиентной интерполяции;

Reviewer: ```suggestion
- В Photoshop [был добавлен](https://helpx.adobe.com/photoshop/using/gradient-interpolation.html) Oklab для градиентов;
```

- OKlab стали использовать в [генераторах палитр](https://huetone.ardov.me/) для улучшения доступности.

> Уверены: грядущие изменения в CSS Colors 4 и 5 знаменуют момент, когда мы сможем выбрать единственное лучшее решение. Решение, которое позволит создать экосистему на основе нововведений из спецификаций CSS.

Reviewer: Неа, тут смысл предложения в другом:

```suggestion
Не важно что OKLCH молодой. Всё равно у CSS Colors 4 и 5 столько изменений, что экосистему разраотки придётся сильно переделывать. Мы верим, что раз мы начинаем сначала, то лучше взять самое современное и эффективное решение.
```

Author: Мне нравится.
Только может, немного изменить второе предложение? "Сильно переделывать экосистему" субъективно показалось перебор разговорным.

`Всё равно у CSS Colors 4 и 5 столько изменений, что экосистему разработки придётся сильно переделывать.` ->
`Всё равно грядущие изменения в CSS Colors 4 и 5 потребуют сильных перемен в экосистеме разработки.`

Reviewer: Ага, хорошее предложение

#### Оси

Цвета в OKLCH записываются четырьмя числами. В CSS это выглядит так: `oklch(L C H)` или `oklch(L C H / a)`.

<figure>
    <img src="images/oklch-axes.avif" loading="lazy" alt="4 столбца, обозначающие оси OKLCH. Верхний слева означает яркость; он начинается как тёмно-фиолетовый, становится светло-фиолетовым при движении вправо, и в конечном итоге становится малоотличимым от белого. Верхний справа столбец обозначает насыщенность. Он начинается как серовато-фиолетовый и переходит в светло-фиолетовый, начиная с трети столбца и в итоге едва отличается от белого цвета. Нижний слева столбец обозначает альфу: переход альфа-канала в виде оттенков фиолетового, начиная с едва заметных. Нижний справа столбец означает оттенок: вариации ярких цветов, которые сменяются друг другом.">
    <figcaption>Оси OKLCH.</figcaption>
</figure>

Объясним значение каждого компонента более детально:

> - `L` — это воспринимаемая яркость. Варьируется от `0%` (чёрный) до `100%` (белый). Указывать `%` необходимо даже для нулевых значений;

Reviewer: ```suggestion
- `L` — воспринимаемая яркость. Варьируется от `0%` (чёрный) до `100%` (белый). Указывать `%` необходимо даже для нулевых значений;
```

Из Википедии у меня традиция упускать `это` после тире, так как на письмо и так всё понятно

Author: Учту. Хотя иногда замечаю, что мне хочется видеть `это` в качестве зацепки для глаза. При беглом чтении помогает связать разные части предложения, но это субъективно

> - `C`  — это насыщенность, интенсивность цвета. Варьируется от `0` (серый) до бесконечности. На практике у насыщенности всё же есть максимальное значение, но оно зависит от поддерживаемой цветовой гаммы (например, для P3 она больше, чем для sRGB). Каждый оттенок имеет разную максимальную насыщенность. Для обоих наборов P3 и sRGB значение обычно всегда меньше `0.37`.

Reviewer: ```suggestion
- `C`  — это насыщенность, интенсивность цвета. Варьируется от `0` (серый) до бесконечности. На практике у насыщенности всё же есть максимальное значение, но оно зависит от поддерживаемой цветового охвата (например, для P3 она чуть больше, чем для sRGB). Каждый оттенок имеет разную максимальную насыщенность. Но и в P3 и sRGB значение всегда меньше `0.37`.
```

> - `H` — оттенок, угол поворота от `0` до `360`. Красный имеет значение `0`, жёлтый `70`, зелёный `120`, голубой `200`, фиолетовый `300` и затем вновь идут значения красного. Так как это угол, то [`0`](https://oklch.evilmartians.io/#70,0.1,0,100) и [`360`](https://oklch.evilmartians.io/#70,0.1,360,100) означают одинаковый оттенок. `H` можно записать как с единицей измерения (`60deg`), так и без неё (`60`).

Reviewer: Красный это не только `20`, но и вокруг

```suggestion
- `H` — оттенок, угол поворота от `0` до `360`. В районе `0` — красный, `70` — жёлтый, `120` — зелёный, `200` — голубой, `300` — фиолетовый и затем вновь идут значения красного. Так как это угол, то [`0`](https://oklch.evilmartians.io/#70,0.1,0,100) и [`360`](https://oklch.evilmartians.io/#70,0.1,360,100) означают одинаковый оттенок. `H` можно записать как с единицей измерения (`60deg`), так и без неё (`60`).
```

  <span class="property">color:&nbsp;</span><div class="preview-with-value" style="--device-width: 280px; --mobile-width: 205px;"><div class="color-preview without-opacity" style="background-color: rgb(197, 191, 154);"></div><span class="value">oklch(80% 0.05 100);</span></div><span class="comment">/* серовато-жёлтый */</span>
  <span class="property">color:&nbsp;</span><div class="preview-with-value" style="--device-width: 280px; --mobile-width: 205px;"><div class="color-preview without-opacity" style="background-color: rgb(88, 206, 248);"></div><span class="value">oklch(80% 0.12 225);</span></div><span class="comment">/* голубой, с той же воспринимаемой яркостью */</span>
<span>}</span>
<span class="selector">.opacity&nbsp;</span><span>{</span>
   <span class="property">color:&nbsp;</span><div class="preview-with-value"><div class="color-preview" style="background-color: rgba(208, 191, 94, 0.5);"></div><span class="value">oklch(80% 0.12 100 / 50%);</span></div><span class="comment"> /* прозрачный жёлтый */</span>
<span>}</span>
</code>
</pre>

> Стоит помнить, что некоторые компоненты могут содержать [`none`](https://www.w3.org/TR/css-color-4/#valdef-color-none) в значениях. Например, у белого цвета нет оттенков, и браузеры будут трактовать `none` как `0`:

Reviewer: ```suggestion
Стоит помнить, что некоторые компоненты могут содержать [`none`](https://www.w3.org/TR/css-color-4/#valdef-color-none) в значениях. Например, у белого цвета нет оттенков и некоторые конверторы поставят в оттенке `none`. Браузеры будут трактовать `none` как `0`.
```

<pre data-lang="css">
<code tabindex="0" class="language-css">
<span class="selector">:root&nbsp;</span><span>{</span>
  <span class="property">--origin:&nbsp;</span><div class="preview-with-value"><div class="color-preview without-opacity" style="background-color: #0f0;"></div><span class="value">#ff000;</span></div>
<span>}</span>
<span class="selector">.foo&nbsp;</span><span>{</span>
  <span class="property">color:&nbsp;</span><div class="preview-with-value"><div class="color-preview without-opacity" style="background-color: #0f0;"></div><span class="value">oklch(from var(--origin) l c h);</span></div>
<span>}</span>
</code>
</pre>
Начальный цвет (`var(--origin)` в примере выше) может являться:

- Цветом в любом формате: `#ff0000`, `rgb(255, 0, 0)`, или `oklch(62.8% 0.25 30)`.
- Кастомным свойством с цветом в любом формате.

Любой компонент (`l`, `c`, `h`) после `from X` может быть:

- Буквой (`l`, `c`, `h`), которая укажет сохранить компонент таким, каким он был для начального цвета;
- Выражением `calc()`. Вы можете использовать буквы (`l`, `c`, `h`) вместо числа, чтобы сослаться на значение в начальном цвете;
- Новым значением, которое заменит компонент.

> Это может прозвучать сложно, но примеры ниже должны внести ясность:

Reviewer: Прошлое время подчеркнёт, что речь про текст выше

```suggestion
Это может звучать сложно, но примеры ниже должны внести ясность:
```

<span>}</span>
<span class="selector">.message.is-error&nbsp;</span><span>{</span>
  <span class="comment">/* Тот же цвет, но с другой прозрачностью */</span>
  <span class="property">background:&nbsp;</span><div class="preview-with-value"><div class="color-preview" style="background-color: rgba(206, 83, 66, 0.6);"></div><span class="value">oklch(from var(--origin) l c h);</span></div>
  <span class="comment">/* На 10% темнее */</span>
  <span class="property">border-color:&nbsp;</span><div class="preview-with-value"><div class="color-preview without-opacity" style="background-color: rgb(172, 50, 37);"></div><span class="value">oklch(from var(--error) calc(l - 10%) c h);</span></div>
<span>}</span>
<span class="selector">:message.is-success&nbsp;</span><span>{</span>
  <span class="comment">/* Другой оттенок зелёного с той же яркостью и насыщенностью */</span>
  <span class="property">background:&nbsp;</span><div class="preview-with-value"><div class="color-preview without-opacity" style="background-color: rgb(68, 150, 48);"></div><span class="value">oklch(from var(--error) l c 140);</span></div>
<span>}</span>
</code>
</pre>

> Предсказуемость контраста в OKLCH будет полезна при генерации акцентирующих цветов через пользовательский ввод. Попробуйте проверить этот пример в [цветовом миксере](https://oklch.evilmartians.io/#70,0.1,319,100) Злых марсиан:

Reviewer: ```suggestion
Предсказуемость контраста в OKLCH будет полезна, чтобы сделать цвет UI из произвольного цвета, который задаёт пользовать. Посмотрите как выбранный цвет влияет на цвета чекбоксов в [цветовом миксере](https://oklch.evilmartians.io/#70,0.1,319,100) Злых марсиан:
```

<pre data-lang="css">
<code tabindex="0" class="language-css">
<span class="selector">:root&nbsp;</span><span>{</span>
  <span class="comment">/* Яркость и насыщенность заменены на определённый контраст */</span>
  <span class="property">--accent:&nbsp;</span><div class="preview-with-value"><span class="value">oklch(from (--user-input) 87% 0.06 h);</span></div>
<span>}</span>
<span class="selector">body&nbsp;</span><span>{</span>
  <span class="property">background:&nbsp;</span><div class="preview-with-value"><span class="value">var(--accent);</span></div>
  <span class="comment">/* Нам не нужно указывать цвет текста с color-contrast(),
   потому что OKLCH имеет предсказуемый контраст.
   Любой фон с L≥87% имеет хороший контраст с чёрным текстом.*/</span>
  <span class="property">color:&nbsp;</span><div class="preview-with-value"><span class="value">black;</span></div>
<span>}</span>
</code>
</pre>

#### Гамма-коррекция

> У OKLCH есть и другая замечательная особенность: независимость от устройств. OKLCH создан не только для обычных мониторов с sRGB-набором.

Reviewer: ```suggestion
У OKLCH есть и другая замечательная особенность: независимость от устройств. OKLCH создан не только для обычных мониторов с набром цветов sRGB.
```

Мы можем указать абсолютно любой цвет с помощью OKLCH: sRGB, P3, Rec2020 и далее. Ряд комбинаций компонентов поддерживается только на P3-мониторах; а для некоторых сочетаний ещё предстоит изобрести экраны, которые смогут их отображать.

> Не пугайтесь выхода за пределы возможностей экрана — браузеры умеют искать и отображать максимально похожий цвет. Процесс поиска наиболее похожего цвета в другой гамме называется гамма-коррекцией.

Reviewer: ```suggestion
Не пугайтесь выхода за пределы возможностей экрана — браузеры найдут максимально похожий цвет. Процесс поиска наиболее похожего цвета в другом охвате называется гамма-коррекцией.
```

gamut — это охват

Author: Точно, спасибо. Читала ведь про разницу gamut и space, а в этом предложении спутала с третьим термином)

По этой причине в [цветовом миксере OKLCH](https://oklch.evilmartians.io/#60,0.16,140,100) можно увидеть бреши: каждый оттенок имеет разную максимальную насыщенность. Эта  проблема связана не с кодированием цвета в OKLCH, а с пределами возможностей существующих мониторов и нашего зрения.

> Например, при некоторой яркости только у голубого цвета может быть большая насыщенность; или для зелёного цвета может не оказаться соответствующего голубого или красного с той же насыщенностью. 

Reviewer: ```suggestion
Например, при некоторой яркости только у голубого цвета может быть самая большая насыщенность — в зелёном или красном оттенке не будет цвета с той же насыщенностью. 
```

Author: Мне казалось, что _в цвете есть оттенок_, чем _в оттенке есть цвет_. Можешь, пожалуйста, объяснить выбор слов здесь?

Reviewer: У меня тут логика такая. Цвет — это конкретное значение H, L, C. Оттенок — это какое-то значение H. Да, получается в цвете есть оттенок.

Но можно посмотреть и с другой стороны — у какого-то конкретного H есть куча комбинаций с L и C. Так что внутри оттенка куча цветов с тем же H.

Author: Поняла, спасибо.

<figure>

>     <img src="images/oklch-l-44.avif" loading="lazy" alt="График с двумя осями, C слева и H снизу. Он показывает фигуру, на вершине которой есть большие круглые пустоты. Слева направо на фигуре сменяются различные цвета, для которых показаны градиентные переходы.">

Reviewer: Лучше сфокусируйся на пиках и объясни, что пик зелёного выше остальных пиков

Author: Голубого ведь?

Поправила на `График с двумя осями, C по горизонтали и H по вертикали. На горизонтальной шкале графика оттенки цветов сменяются слева направо. Пиковые значения на вертикальной шкале показывают, что при изменении яркости разные оттенки обладают разной максимальной насыщенностью. Пик насыщенности для голубого заметно выше, чем для других цветов.`

Стало ли лучше?

Reviewer: > Голубого ведь?

Ой, да. Точнее синего.

Reviewer: > Стало ли лучше?

Ага. Стоит изменить на синий. Или есть логика где голубой, где синий?

Author: Да, здесь, наверное, правильнее сказать именно про "синий"
![Screenshot from 2022-12-08 16-24-55](https://user-images.githubusercontent.com/78731462/206445907-f0636bed-8d4e-4a77-8f38-3d0ca2618a12.png)
Поправила

    <figcaption>Для 44% яркости только голубой цвет имеет насыщенность, заметную на sRGB-экранах.</figcaption>
</figure>

Есть два способа гамма-коррекции:

> - Конвертировать цвет в RGB или P3 и отсечь значения более 100% или менее 0%: `rgb(150% -20% 30%)` → `rgb(100% 0 30%)`. Это наиболее быстрый способ, но у него наихудший результат — он заметно может изменить оттенок цвета;

Reviewer: ```suggestion
- Конвертировать цвет в RGB или P3 и отсечь значения более 100% или менее 0%: `rgb(150% -20% 30%)` → `rgb(100% 0 30%)`. Это наиболее быстрый способ, но у него наихудший результат — он может заметно изменить оттенок цвета;
```

- Перевести цвет в формат OKLCH и уменьшить насыщенность и яркость. Это сохранит тот же оттенок, но немного замедлит отрисовку.

Крис Лиллей создал [интересное сравнение](https://svgees.us/Color/ok-clip-lch-explorer.html) между разными методами гамма-коррекции.

> Спецификация CSS Colors 4 [требует](https://www.w3.org/TR/css-color-4/#css-gamut-mapping) браузеры применять метод OKLCH для гамма-коррекции. Однако прямо сейчас Safari использует быстрый, но неточный метод отсечения.

Reviewer: ```suggestion
Спецификация CSS Colors 4 [требует](https://www.w3.org/TR/css-color-4/#css-gamut-mapping) от браузеров применять OKLCH-метод для гамма-коррекции. Однако прямо сейчас Safari использует быстрый, но неточный метод отсечения.
```

> Именно поэтому мы рекомендуем ручную гамма-коррекцию, а также указывать как sRGB, так и P3-цвета:

Reviewer: ```suggestion
Именно поэтому мы рекомендуем ручную гамма-коррекцию — указывать как sRGB, так и P3-цвета:
```

Это объяснение смысла ручного, а не дополнительная информация

<pre data-lang="css">
<code tabindex="0" class="language-css">
<span class="selector">.martian&nbsp;</span><span>{</span>
 <span class="property">background:&nbsp;</span><div class="preview-with-value"><div class="color-preview without-opacity" style="background-color: rgb(159, 166, 1);"></div><span class="value">oklch(69.73% 0.155 112.79);</span></div>
<span>}</span>
<span class="selector">@media (color-gamut: p3)&nbsp;</span><span>{</span>
  <span class="selector">.martian&nbsp;</span><span>{</span>
    <span class="property">background:&nbsp;</span><div class="preview-with-value"><div class="color-preview without-opacity" style="background: color(display-p3 0.6327 0.6542 0.0677);"></div><span class="value">oklch(69.73% 0.176 112.79);</span></div>
    <span class="comment">/* Превью этого цвета вы сможете увидеть только на P3-мониторе в Safari */</span>
  <span>}</span>
<span>}</span>
</code>
</pre>

> И здесь есть хорошая новость: полифил `oklch` умеет это делать. Он автоматически оборачивает P3-цвета правилом `@media` и устанавливает в качестве запасного варианта ближайший sRGB-цвет.

Reviewer: ```suggestion
И здесь есть хорошая новость: полифил `oklch()` умеет это делать. Он автоматически оборачивает P3-цвета выражением `@media` и устанавливает в качестве запасного варианта ближайший sRGB-цвет.
```

https://github.com/web-standards-ru/dictionary/blob/main/dictionary.md#m

- Конвертировать цвет в RGB или P3 и отсечь значения более 100% или менее 0%: `rgb(150% -20% 30%)` → `rgb(100% 0 30%)`. Это наиболее быстрый способ, но у него наихудший результат — он может заметно изменить оттенок цвета;
- Перевести цвет в формат OKLCH и уменьшить насыщенность и яркость. Это сохранит тот же оттенок, но немного замедлит отрисовку.

Крис Лиллей создал [интересное сравнение](https://svgees.us/Color/ok-clip-lch-explorer.html) между разными методами гамма-коррекции.

Спецификация CSS Colors 4 [требует](https://www.w3.org/TR/css-color-4/#css-gamut-mapping) от браузеров применять OKLCH-метод для гамма-коррекции. Однако прямо сейчас Safari использует быстрый, но неточный метод отсечения.

Именно поэтому мы рекомендуем ручную гамма-коррекцию — указывать как sRGB, так и P3-цвета:

<pre data-lang="css">
<code tabindex="0" class="language-css">
<span class="selector">.martian&nbsp;</span><span>{</span>
 <span class="property">background:&nbsp;</span><div class="preview-with-value"><div class="color-preview without-opacity" style="background-color: rgb(159, 166, 1);"></div><span class="value">oklch(69.73% 0.155 112.79);</span></div>
<span>}</span>

> <span class="selector">@media (color-gamut: p3)&nbsp;</span><span>{</span>

Reviewer: Хм, а `&nbsp;` не сломает copy-paste?

Кто их вставляет?

Author: Проверила в текстовых редакторах (google doc и vs code), не сломалось, вместо `&nbsp;` идёт вставка пробела.

Я сама вставила, чтобы гарантировать отсутствие переноса открывающей скобки после названия селектора. Если это кажется избыточным, могу убрать, но мне показалось полезным для самых узких мобилок.

Reviewer: Норм, давай оставим

  <span class="selector">.martian&nbsp;</span><span>{</span>
    <span class="property">background:&nbsp;</span><div class="preview-with-value"><div class="color-preview without-opacity" style="background: color(display-p3 0.6327 0.6542 0.0677);"></div><span class="value">oklch(69.73% 0.176 112.79);</span></div>
    <span class="comment">/* Превью этого цвета вы сможете увидеть только на P3-мониторе в Safari */</span>
  <span>}</span>
<span>}</span>
</code>
</pre>

И здесь есть хорошая новость: полифил `oklch()` умеет это делать. Он автоматически оборачивает P3-цвета выражением `@media` и устанавливает в качестве запасного варианта ближайший sRGB-цвет.

## Внедрение OKLCH в проект

#### Шаг 1: Добавление полифила OKLCH в CSS

> К декабрю  2022, из всех браузеров `oklch()` [поддерживает](https://developer.mozilla.org/en-US/docs/Web/CSS/color_value/oklch) только Safari. К счастью, у нас есть полифил для её статичных значений (но не для кастомных CSS-свойств).

Reviewer: ```suggestion
К декабрю  2022, из всех браузеров `oklch()` [поддерживает](https://developer.mozilla.org/en-US/docs/Web/CSS/color_value/oklch) только Safari (Chrome добавит в версии 110). К счастью, у нас есть полифил для её статичных значений (но не для кастомных CSS-свойств).
```

Релиз Хром 110 уже скоро, кажется лучше сразу добавить.

Reviewer: > К счастью, у нас есть полифил для её статичных значений (но не для кастомных CSS-свойств).

Снова можно неправильно понять, что нельзя `--var: oklch(100% 0 0)`

Author: Может, так, или будет сложно для восприятия:

"К счастью, у нас есть полифил для её статичных значений, например, для `--var: oklch(100% 0 0)` (но не для кастомных CSS-свойств)."

Reviewer: Можем использовать слово динамических, чтобы объяснить разницу

>  К счастью, у нас есть полифил для статичных значений функции. Но цвета с динамическими кастомными свойствами внутри будут всё ещё требовать поддержку браузера.

Есть два полифила, поддерживающих `oklch()`:

- Очень популярный [`postcss-preset-env`](https://preset-env.cssdb.org/);
- И крайне быстрый [Lightning CSS](https://lightningcss.dev/), написанный на Rust.

Скорее всего, в вашей системе уже есть `postcss-preset-env`; например, он есть у каждого проекта, созданного с помощью Create React App.

> 1. Проверьте ваш файл блокировки (`package-lock.json`, `yarn.lock` или `pnpm-lock.yaml`) для `postcss-preset-env`. Убедитесь, что он использует последнюю версию: `7.x` или выше;

Reviewer: ```suggestion
1. Проверьте ваш лок-файл (`package-lock.json`, `yarn.lock` или `pnpm-lock.yaml`) для `postcss-preset-env`. Убедитесь, что он использует версию: `7.x` или выше;
```

Боюсь, «файл блокировки» не устоялось и люди не поймут (я бы вообще назвал «файл заморозки/фиксации», так как там не про блокирование, а не фиксацию). Давай пока транслитерируем.

А `последнюю` версию убрал, так как не факт что она будет такой на момент выхода статьи. Да и от убирания слова смысл не меняется.

> 2. Добавьте `oklch(100% 0 0)` в ваш CSS и посмотрите, как он скомпилируется в `rgb()` с помощью инструмента сборки.

Reviewer: ```suggestion
2. Добавьте `oklch(100% 0 0)` в ваш CSS и посмотрите, скомпилируется ли сборка его в `rgb()`.
```

Если у вас нет `postcss-preset-env`, но есть инструмент для сборки фронтенда (такой как webpack):

1. Установите `postcss-preset-env` с помощью пакетного менеджера. Для npm запустите следующую команду:
<pre data-lang="sh">
<code tabindex="0" class="language-sh">
npm install postcss-preset-env postcss
</code>
</pre>

> 2. Проверьте [документацию PostCSS](https://github.com/postcss/postcss), чтобы узнать, как добавить его поддержку в ваш инструмент сборки. Например webpack требует [`postcss-loader`](https://github.com/webpack-contrib/postcss-loader), а в [Vite](https://vitejs.dev/) уже есть встроенная поддержка.

Reviewer: > как добавить его поддержку

Не понятно что тут `его` — полфил или PostCSS (правильный ответ).

Author: Сначала так и написала (`PostCSS` вместо `его`), но что-то покоробила тавтология(

Author: Изменила `его` на `PostCSS`, не уверена насчёт этого момента

> 3. Если у вас уже есть интеграция с PostCSS, найдите файл с его конфигурацией. Многие проекты уже используют PostCSS (например, Autoprefixer). В корневой папке проекта найдите `postcss.config.js` или `.postcssrc.json`, а затем раздел "postcss" в `package.json` или `postcss` в конфигурации сборщика.

Reviewer: Стоит попробовать написать `конфиг`?

Reviewer: `Autoprefixer` → `Автопрефиксер`

Reviewer: `а затем раздел` → `или раздел`

Author: Поменяла на Автопрефиксер +

`Конфиг` звучит жаргонно, на мой взгляд(

Если "а затем" поменять на "или", то будет сразу три "или" в одном предложении. Может, тогда так:
_В корневой папке проекта найдите `postcss.config.js` (`.postcssrc.json`), или раздел "postcss" в `package.json` или конфигурации сборщика._

Reviewer: > В корневой папке проекта найдите postcss.config.js (.postcssrc.json), или раздел "postcss" в package.json или конфигурации сборщика.

Отлично

4. Если вы смогли найти файл конфигурации PostCSS, добавьте `postcss-preset-env` в плагины:
<pre data-lang="css">
<code tabindex="0" class="language-css">
{
  "plugins": [
+   "postcss-preset-env",
    "autoprefixer"
  ]
}
</code>
</pre>
5. Если вы не смогли найти файл конфигурации, создайте файл `.postcssrc.json` в корневой папке проекта:
<pre data-lang="css">
  <code tabindex="0" class="language-css">
  {
    "plugins": [
      "postcss-preset-env"
    ]
  }
  </code>
</pre>
Если у вас ещё нет инструмента сборки, мы рекомендуем использовать Vite или компилировать CSS с помощью [lightningcss](https://github.com/parcel-bundler/lightningcss#from-the-cli) CLI.

> Добавьте в файл `.test{ background: oklch(100% 0 0) }`, скомпилируйте стили, затем найдите в итоговом CSS `.test{background:#fff}`.

Reviewer: ```suggestion
Добавьте в файл `.test{ background: oklch(100% 0 0) }`, скомпилируйте стили. Если в итоговом CSS есть `.test{background:#fff}`, то всё работает.
```

#### Шаг 2: Перевод в OKLCH уже существующих цветов

> С помощью полифила `oklch()` мы можем заменить все цвета в форматах hex, `rgb()` или `hsl()` на OKLCH. Они будут работать в любом бразуере.

Reviewer: ```suggestion
После установки полфила `oklch()`, вы можете заменить все цвета в форматах hex, `rgb()` или `hsl()` на OKLCH. Они будут работать в любом бразуере.
```

Найдите любой цвет в вашем исходном CSS-коде и переведите его в `oklch()`, используя цветовой миксер [OKLCH](https://oklch.evilmartians.io/#70,0.1,191,100).
<pre data-lang="css">
  <code tabindex="0" class="language-css">
  .header {
  - background: #f3f7fa;
  + background: oklch(97.4% 0.006 240);
  }
  </code>
</pre>
Вы также можете использовать [этот скрипт](https://github.com/fpetrakov/convert-to-oklch), чтобы автоматически перевести все цвета:
<pre data-lang="sh">
  <code tabindex="0" class="language-sh">
  npx convert-to-oklch ./src/**/*.css
  </code>
</pre>

> Помните, что эти полифилы поддерживаются только в CSS, и не будут работать в HTML или JS-файлах.

Reviewer: ```suggestion
Помните, что эти полифилы поддерживают только CSS, и не будут работать в HTML или JS-файлах.
```

#### Дополнительно: добавление палитры

> Небольшой рефакторинг заметно улучшит поддерживаемость CSS-кода. Давайте попробуем вынести цвета в единую палитру?

Reviewer: ```suggestion
Раз мы рефакторим CSS, то может стоит воспользоваться моментом и ещё и внедрить общую палитру?
```

Author: Будет ли так проще и "ближе к читателю"?

"Раз мы уже рефакторим CSS, то может, воспользуемся моментом и заодно внедрим общую палитру?"

Reviewer: > Раз мы уже рефакторим CSS, то может, воспользуемся моментом и заодно внедрим общую палитру?

Да, тоже хорошо

> Наши требования для палитр:

Reviewer: ```suggestion
Что такое общая палитра:
```

Author: До этого в предложении используется термин "_единая_". Если мы его поясняем дальше, то может, оставить в исходном виде и не менять на синоним?

Reviewer: Вот «единая палитра» мне кажется не таким понятным, как «общая палитра». Тут слово «общее» означает что её можно использовать в разных местах приложений («общяя для всех»).

Author: Соглашусь

- Все цвета указываются как кастомные CSS-свойства;
- Компоненты фреймворков используют эти цвета в виде `var(--error)`;

> - Дизайнерам стоит стремиться к переиспользованию цветов. Это позволит уменьшить количество их возможных сочетаний;

Reviewer: ```suggestion
- Дизайнерам стоит стремиться к переиспользованию цветов.
```

Там речь не про сочетания. Но я не знаю как перевести — можно просто отбросить второе предложение.

> - Для создания тёмной темы или темы с высоким контрастом не нужны выражения `@media` в компонентах. Должно быть достаточно изменения кастомных свойств в палитре.

Reviewer: ```suggestion
- Тёмная тема или тема с высоким контрастом будет делаться через переопределение кастомных свойств в палитре, а не через `@media` в каждом компонентах
```

Author: До этого пункты в списке описываются в настоящем времени, как требования. Предлагаю здесь тоже оставить настоящее время:

"Тёмная тема или тема с высоким контрастом создаются через переопределение кастомных свойств в палитре, а не через `@media` в каждом компоненте."

Reviewer: > Тёмная тема или тема с высоким контрастом создаются через переопределение кастомных свойств в палитре, а не через @media в каждом компоненте.

👍

<code tabindex="0" class="language-css">
<span class="selector">@media (prefers-color-scheme: dark)&nbsp;</span><span>{</span>
  <span class="selector">:root&nbsp;</span><span>{</span>
    <span class="property">--surface-0:&nbsp;</span><div class="preview-with-value"><span class="value">oklch(0% 0 0);</span></div>
    <span class="property">--surface-1:&nbsp;</span><div class="preview-with-value"><span class="value">oklch(29% 0.01 300);</span></div>
    <span class="property">--surface-2:&nbsp;</span><div class="preview-with-value"><span class="value">oklch(29% 0 0 / 85%);</span></div>
    <span class="property">--text-primary:&nbsp;</span><div class="preview-with-value"><span class="value">oklch(100% 0 0);</span></div>
  <span>}</span>
<span>}</span>
</code>
</pre>
Также отметим, что перейти к `oklch()` станет проще после создания единой палитры.

#### Шаг 3: Поддержка OKLCH с помощью Stylelint

[Stylelint](https://stylelint.io/) — статический анализатор стилей, который будет полезен для поиска распространённых ошибок и внедрения хороших практик. Он работает как ESLint, но для CSS, SASS, или CSS-in-JS.

Stylelint будет очень полезен при переходе на `oklch()`:

> - Он позволяет указать, что цвета в форматах hex, `rgb()` и `hsl()` не будут использоваться. Следует хранить все цвета в `oklch()` для улучшения единообразия;

Reviewer: ```suggestion
- Он напомнит коллегам, что цвета в форматах hex, `rgb()` и `hsl()` не будут использоваться. Следует хранить все цвета в `oklch()` для улучшения единообразия;
```

- Он проверит, что все P3-цвета находятся в выражении `@media (color-gamut: p3)`. Это поможет избежать автоматической гамма-коррекции в браузере (Safari делает её неправильно).

> Давайте установим Stylelint и плагин [stylelint-gamut](https://www.npmjs.com/package/stylelint-gamut). Воспользуемся пакетным менеджером npm:

Reviewer: ```suggestion
Давайте установим Stylelint и плагин [stylelint-gamut](https://www.npmjs.com/package/stylelint-gamut). Для npm:
```

<pre data-lang="sh">
  <code tabindex="0" class="language-sh">
  npm install stylelint stylelint-gamut
  </code>
</pre>
Затем создадим конфигурацию `.stylelintrc` следующим образом:
<pre data-lang="json">
  <code tabindex="0" class="language-json">
  {
    "plugins": [
      "stylelint-gamut"
    ],
    "rules": {
      "gamut/color-no-out-gamut-range": true,
      "function-disallowed-list": ["rgba", "hsla", "rgb", "hsl"],
      "color-function-notation": "modern",
      "color-no-hex": true
    }
  }
  </code>
</pre>
Добавим вызов Stylelint в команду `npm test`, чтобы запустить его в CI. Изменим следующую строку в `package.json`:
<pre data-lang="json">
  <code tabindex="0" class="language-json">
    "scripts": {
  -   "test": "eslint ."
  +   "test": "eslint . && stylelint **/*.css"
    }
  </code>
</pre>
Запустим `npm test`, чтобы увидеть, какие цвета должны быть переведены в `oklch()`.

> Мы также рекомендуем добавить [`stylelint-config-recommended`](https://github.com/stylelint/stylelint-config-recommended) в `.stylelintrc`. Эта рекомендуемая конфигурация упростит проверку хороших практик в CSS-коде.

Reviewer: ```suggestion
Мы также рекомендуем добавить [`stylelint-config-recommended`](https://github.com/stylelint/stylelint-config-recommended) в `.stylelintrc`. Это набор самых популярных практик в CSS.
```

Reviewer: «рекомендуемая конфигурация» — слишком официально

#### Дополнительно: P3-цвета

> Перевод цветов в `oklch()` улучшит читаемость и поддерживаемость кода, но не добавит новых возможностей. Тем не менее, у OKLCH есть заметное пользователям преимущество: с помощью него мы можем добавлять на сайты красочные и глубокие P3-цвета.

Reviewer: Что-то слишком сложно и далеко от обычной речи.

Может так:

```suggestion
Просто переведя цвета в `oklch()` вы улучшите читаемость и поддерживаемость кода, но не добавите что-то заметное для пользователей. А вот если ещё и внедрить красочные и глубокие P3-цвета в дизайн, то и пользователи смогут оценить рефакторинг.
```

_P3-цвета очень полезны при создании палитр. Но прямо сейчас мы не можем их использовать: только Safari на современных устройствах Apple поддерживает P3-цвета._

Вот, как можно добавить P3-цвета в проект:

> 1. Выберите любой насыщенный цвет (например, один из акцентирующих) в вашем CSS-коде;

Reviewer: Не уверен, что люди используют слово `акцентирующих`. «Главных»?

Author: Не помню, чтобы я когда-нибудь говорила "главные цвета". Может, совсем убрать пояснение в скобках? Мне кажется, и так понятно, что такое насыщенный цвет.

Reviewer: Да, давай

2. Скопируйте его и вставьте в [цветовой миксер OKLCH](https://oklch.evilmartians.io/);

> 3. Измените значения `Chroma` и `Lightness` так, чтобы цвет оказался в области P3-цветов. Легче всего это сделать, изменив `Lightness`: выберите любой цвет выше тонкой белой линии на этом графике;

Reviewer: ```suggestion
3. Измените значения `Chroma` и `Lightness` так, чтобы цвет оказался в области P3-цветов. Легче всего это сделать, смотря на график `Lightness`: выберите любой цвет выше тонкой белой линии на этом графике;
```

4. Скопируйте результат и оберните его в медиавыражение `color-gamut: p3`.
<pre data-lang="css">
<code tabindex="0" class="language-css">
<span class="selector">:root&nbsp;</span><span>{</span>
  <span class="property">--accent:&nbsp;</span><div class="preview-with-value"><div class="color-preview without-opacity" style="background-color: rgb(48, 189, 68)"></div><span class="value">oklch(70% 0.2 145);</span></div>
<span>}</span>
<span class="selector">@media (color-gamut: p3)&nbsp;</span><span>{</span>
  <span class="selector">:root&nbsp;</span><span>{</span>
    <span class="property">--accent:&nbsp;</span><div class="preview-with-value"><div class="color-preview without-opacity" style="background-color: color(display-p3 0.1530 0.7673 0.0793);"></div><span class="value">oklch(70% 0.29 145);</span></div>
  <span>}</span>
<span>}</span>
</code>
</pre>

#### Дополнительно: OKLCH для SVG

> OKLCH можно использовать не только в CSS, но и в SVG или HTML. Это полезно при добавлении ярких красок для иконок приложения.

Reviewer: ```suggestion
OKLCH можно использовать не только в CSS, но и в SVG или HTML. Это полезно при добавлении ярких красок в иконки.
```

> Обратите внимание: полифилов для использования SVG нет. По этой причине мы рекомендуем использовать функцию `oklch()` только при указании P3-цветов.

Reviewer: ```suggestion
Обратите внимание: полифилов для SVG нет. По этой причине мы рекомендуем использовать `oklch()` только для P3-цветов.
```

<pre data-lang="xml">
<code tabindex="0" class="language-xml">
&lt;svg viewBox="0 0 120 120" xmlns="http://www.w3.org/2000/svg"&gt;
  <div>&nbsp;&nbsp;&lt;style&gt;
    @media (color-gamut: p3) {
      rect {
      &nbsp;&nbsp;fill:&nbsp;<div class="color-preview without-opacity" style="background-color: color(display-p3 0.0505 0.5557 0.0653);"></div>oklch(55% 0.23 146)
      }
    }
  &lt;/style&gt;</div>
  <div>&nbsp;&nbsp;&lt;rect x="10" y="10" width="100" height="100"
          fill="&nbsp;<div class="color-preview without-opacity" style="background-color: #048c2c;"></div>#048c2c" /&gt;</div>
&lt;/svg&gt;
</code>
</pre>
#### Дополнительно: OKLCH и Oklab в градиентах

Градиент — путь между двумя или более точками цветового пространства. В разных пространствах градиент для одних и тех же цветов может получиться совершенно разный.
<figure>

>     <img src="images/p3.png" loading="lazy" alt="4 квадрата. В каждом квадрате приведён код градиентов в пространствах sRGB, Oklab и OKLCH. Код в каждом блоке одинаков, но для каждого цветового пространства градиент немного различается, что показывают фоновые цвета квадратов.">

Reviewer: ```suggestion
    <img src="images/p3.png" loading="lazy" alt="4 блока. В каждом блока приведён код градиентов в пространствах sRGB, Oklab и OKLCH. Начальный и конечный цвет в каждом блоке одинаков, но для каждого цветового пространства градиент немного различается, что показывают фоновые цвета квадратов.">
```

Может добавим этот код?

Author: Наверное лучше добавить, хоть и громоздко; добавила.

    <figcaption>Так выглядят градиенты в разных цветовых пространствах.</figcaption>
</figure>

> Нет единого решения, какое пространство лучше для работы с градиентами; это зависит от конкретной задачи. Но в цветовом пространстве Oklab (родственнике OKLCH, живущем на вершине декартовых координат) обычно получается хороший результат:

Reviewer: ```suggestion
Нет единого решения, какое пространство лучше для работы с градиентами; это зависит от конкретной задачи. Но в цветовом пространстве Oklab (родственнике OKLCH, живущем в декартовых координат) обычно получается хороший результат:
```

- У него нет сероватой области в центре, как в sRGB (пространство, используемое по умолчанию);
- У него нет сдвига от голубого к фиолетовому, в отличие от Lab;

В [CSS Image 4 specification](https://drafts.csswg.org/css-images-4/#linear-gradients) есть специальное выражение, изменяющее цветовое пространство для градиентов:
<pre data-lang="css">
<code tabindex="0" class="language-css">
<span class="selector">.oklch&nbsp;</span><span>{</span>
  <span class="property">background:&nbsp;</span><div class="preview-with-value"><span class="value">linear-gradient(in lab, blue, green);</span></div>
<span>}</span>
</code>
</pre>
Код выше будет работать только в Safari Technological Preview. Вы можете использовать [плагин PostCSS](https://github.com/csstools/postcss-plugins/tree/main/plugins/postcss-gradients-interpolation-method), чтобы применять его прямо сейчас.

> _Вы можете сделать градиенты ещё более красивыми, используя [easing-gradients](https://larsenwork.com/easing-gradients/)._

Reviewer: ```suggestion
_С [easing-gradients](https://larsenwork.com/easing-gradients/) градиенты получаются ещё более красивыми._
```

#### Дополнительно: Изменение цветов с помощью OKLCH

Уже очень скоро мы сможем увидеть истинную силу OKLCH: с [CSS Colors 5](https://www.w3.org/TR/css-color-5/#relative-colors) браузеры начнут поддерживать встроенные изменения цветов в CSS.

OKLCH крайне хорош для работы с цветом. В отличие от HSL, его контраст предсказуем; в нём нет сдвига оттенка при изменении насыщенности, чем не может похвастаться LCH.

> К сожалению, полифил не даёт полную поддержку кастомных CSS-свойств. Но если мы познакомимся с OKLCH сейчас, то сможем применять знания в реальных проектах, как только это станет возможным.

Reviewer: ```suggestion
К сожалению, полифилы не помогут нам тут — нужно ждать поддержки браузеров. Но если мы познакомимся с OKLCH сейчас, то вы будете уже готовы, когда в браузерах появится поддержка изменения цвета.
```

Author: _Тут_ кажется сильно разговорным.

Может, "К сожалению, здесь нам полифилы не помогут — нужно ждать поддержки браузеров..."?

Reviewer: Да, давай

Следующий пример иллюстрирует, как можно при наведении на кнопку сделать её фон на 10% темнее:
<pre data-lang="css">
<code tabindex="0" class="language-css">
<span class="selector">.button&nbsp;</span><span>{</span>
  <span class="property">background:&nbsp;</span><div class="preview-with-value"><div class="color-preview without-opacity" style="background-color: rgb(165, 145, 213);"></div><span class="value">var(--accent);</span></div>
<span>}</span>
<span class="selector">.button:hover&nbsp;</span><span>{</span>
  <span class="property">background:&nbsp;</span><div class="preview-with-value"><div class="color-preview without-opacity" style="background-color: rgb(135, 115, 181);"></div><span class="value">oklch(from var(--accent) calc(l - 10%) c h);</span></div>
<span>}</span>
</code>
</pre>

> С кастомными свойствами CSS достаточно единожды определить логику`:hover`. В дальнейшем мы можем создавать разные варианты стилей, просто изменяя исходный цвет.

Reviewer: ```suggestion
С кастомными свойствами CSS достаточно единожды определить логику `:hover`. В дальнейшем мы можем создавать разные варианты стилей, просто изменяя исходный цвет.
```

  <span class="property">background:&nbsp;</span><div class="preview-with-value"><span class="value">var(--button-color);</span></div>
<span>}</span>
<span class="selector">.button.is-secondary&nbsp;</span><span>{</span>
  <span class="property">--button-color:&nbsp;</span><div class="preview-with-value"><div class="color-preview" style="background-color: rgba(117, 161, 220, 0.5);"></div><span class="value">var(--dimmed);</span></div>
<span>}</span>
<span class="selector">.button.is-error&nbsp;</span><span>{</span>
  <span class="property">--button-color:&nbsp;</span><div class="preview-with-value"><div class="color-preview" style="background-color: rgb(214, 133, 131);"></div><span class="value">var(--error);</span></div>
<span>}</span>
<span class="selector">.button:hover&nbsp;</span><span>{</span>
  <span class="comment">/* Один :hover для обычного, вторичного и ошибочного состояния */</span>
  <span class="property">background:&nbsp;</span><div class="preview-with-value"><span class="value">oklch(from var(--button-color) calc(l + 10%) c h);</span></div>
<span>}</span>
</code>
</pre>

> Предсказуемый контраст у OKLCH позволяет работать с цветами из пользовательского ввода с отличной доступностью для наших сайтов.

Reviewer: Сложно и люди так не говорят

```suggestion
Поскольку у OKLCH предсказуемый контраст, то мы можем взять случайный цвет от пользователя и сделать из него тему сайта с отличной доступностью:
```

<pre data-lang="css">
<code tabindex="0" class="language-css">
<span class="selector">.header&nbsp;</span><span>{</span>
  <span class="comment">/* JS установит --user-avatar-dominant */</span>
  <span class="property">background:&nbsp;</span><div class="preview-with-value"><div class="color-preview without-opacity" style="background-color: rgb(117, 161, 220);"></div><span class="value">oklch(from var(--user-avatar-dominant) 80% 0.17 h);</span></div>
  <span class="comment">/* OKLCH позволяет быть уверенным, что чёрный текст всегда будет читаемым на любом оттенке, так как мы установили L равным 80% */</span>
  <span class="property">color:&nbsp;</span><div class="preview-with-value"><span class="value">black;</span></div>
<span>}</span>
</code>
</pre>

> И здесь стоит вспомнить про ещё одно достоинство OKLCH: с ним генерация целой системы палитр в CSS становится лёгкой задачей.

Reviewer: ```suggestion
С OKLCH и изменением цветов в будущем можно будет генерировать всю палитру дизайн системы прямо в CSS.
```

Author: Мне нравится, но показалось непонятным, к чему относится "в будущем". 

Как насчёт:
"В будущем, с OKLCH и изменением цветов мы сможем генерировать всю палитру дизайн системы прямо в CSS."

Reviewer: Да, давай

Reviewer: А может стоит внедрить термин «производные цвета»?

В спеке изменения цветов называются relative colors (их можно перевести как относительные, но этот перевод мне кажется неправильный).

Если внедрим в статью такой термин, то тут будет лучше:

> В будущем, с OKLCH и производнысм цветами мы сможем генерировать всю палитру дизайн системы прямо в CSS.

Ну и мы можем стать родоначальником термина в русском языка (надо будет в словарь Веб-стандартов PR тебе послать тоже)

Author: Мне нравится эта идея.
Помимо места в этом предложении, попробовала внедрить термин в следующие места в статье:

1. Уже очень скоро мы сможем увидеть истинную силу OKLCH: с CSS Colors 5 браузеры начнут поддерживать встроенные изменения цветов в CSS. **->  ...начнут поддерживать производные цвета в CSS.**
2. Перейдя на OKLCH сегодня, мы подготовили к себя к завтрашнему дню, когда в CSS появится встроенная работа с цветом. **-> ...когда в CSS появятся производные цвета.**
3. CSS Color 4 — большой шаг в мире стилей, но грядущий CSS Color 5 будет ещё полезнее. В нём, наконец, появятся встроенные изменения цветов. **-> ...В нём, наконец, появятся производные цвета.** 
4. Заголовок "Встроенные изменения цветов в CSS" **-> Производные цвета в CSS** 
5. К сожалению, здесь нам полифилы не помогут — нужно ждать поддержки браузеров. Но если мы познакомимся с OKLCH сейчас, то уже будем готовы, когда в браузерах появится поддержка изменения цвета. **->  ...когда в браузерах появится поддержка производных цветов.**

Всё ли здесь ок?

Reviewer: Нужно в первом применении слова явно сказать, что это изменение цвета встроенное прямо в CSS и оно называется «производные цвета».

И скинь PR в https://github.com/web-standards-ru/dictionary с обоснованием и другими вариантами (с описанием почему не подходят).

Author: Добавила явное пояснение в первом применении.

Открыла PR https://github.com/web-standards-ru/dictionary/pull/467

#### Дополнительно: OKLCH в JS

Нам не нужно ждать CSS Colors 5, чтобы использовать OKLCH для изменений цвета.

> [Color.js](https://colorjs.io/) и [culori](https://culorijs.org/) позволяют менять цвета в JS с учётом преимуществ OKLCH.

Reviewer: ```suggestion
[Color.js](https://colorjs.io/) и [culori](https://culorijs.org/) позволяют менять цвета в JS со всеми преимуществами OKLCH.
```

> Ниже представлен  пример, как сделать акцентирующий цвет с помощью Color.js:

Reviewer: ```suggestion
Ниже представлен  пример, как сгенерировать тему из пользовательского цвета цвет с помощью Color.js:
```

<pre data-lang="js">
  <code class="language-js">
  import Color from 'colorjs.io'

  <span class="comment">// Разбираем любой CSS цвет</span>
  let accent = new Color(userAvatarDominant)

  <span class="comment">/// Устанавливаем яркость и контраст</span>
  accent.oklch.l = 0.8
  accent.oklch.c = 0.17

  <span class="comment">/// Применяем гамма-коррекцию для sRGB, если мы вышли за пределы sRGB</span>
  if (!accent.inGamut('srgb')) {
    accent = accent.toGamut({ space: 'srgb' })
  }

  <span class="comment">/// Делаем цвет на 10% светлее </span>
  let hover = accent.clone()
  hover.oklch.l += 0.1

  document.body.style.setProperty('--accent', accent.to('srgb').toString())
  document.body.style.setProperty('--accent-hover', hover.to('srgb').toString())
  </code>
</pre>

> Попробуйте самостоятельно проверить, как работает culori, используя [исходный код цветового миксера OKLCH](https://github.com/evilmartians/oklch-picker). 

Reviewer: ```suggestion
Как это сделать в culori, можно посмотреть в [исходном коде цветового миксера OKLCH](https://github.com/evilmartians/oklch-picker). 
```

> Вы можете использовать разные библиотеки, чтобы создавать целые системы палитр в цветовом пространстве OKLCH. Они, к тому же, будут обладать предсказуемым констрастом и отличной доступностью. Например, [Hueone](https://huetone.ardov.me/), генератор доступных палитр, по умолчанию использует Oklab.

Reviewer: ```suggestion
Вы можете использовать эти библиотеки, чтобы создавать целые палитр в цветовом пространстве OKLCH. Они будут обладать предсказуемым констрастом и хорошей доступностью. Например, [Hueone](https://huetone.ardov.me/), генератор доступных палитр, по умолчанию использует Oklab.
```

## Подведение итогов: почему мы перешли на OKLCH?

Мы, Злые марсиане, уже используем OKLCH в наших проектах. Ответим на главный вопрос: какие же преимущества мы получили, перейдя на OKLCH?

#### 1. Лёгкое чтение цветов

> Благодаря OKLCH мы без труда определяем записанный цвет, просто посмотрев на числа в CSS-свойстве. И уже в процессе чтения кода мы можем распознать проблему плохого контраста. Для этого нужно всего лишь сравнить яркость в CSS-правилах:

Reviewer: ```suggestion
Благодаря OKLCH мы без труда понимаем что за цвет написан в CSS, просто посмотрев на числа. И уже в процессе чтения кода мы можем распознать проблему плохого контраста. Для этого нужно всего лишь сравнить яркость в CSS-правилах:
```

<span>}</span>
</code>
</pre>
#### 3. Использование как sRGB, так и новых P3-цветов

Мы используем одинаковые CSS-функции как для sRGB, так и для P3-цветов:
<pre data-lang="css">
<code tabindex="0" class="language-css">
<span class="selector">.buy-button&nbsp;</span><span>{</span>
  <span class="property">background:&nbsp;</span><div class="preview-with-value"><div class="color-preview without-opacity" style="background-color: rgb(17, 162, 47);"></div><span class="value">oklch(62% 0.19 145);</span></div>
<span>}</span>
<span class="selector">@media (color-gamut: p3)&nbsp;</span><span>{</span>
  <span class="selector">.buy-button&nbsp;</span><span>{</span>
    <span class="property">background:&nbsp;</span><div class="preview-with-value"><div class="color-preview without-opacity" style="background-color: color(display-p3 0.1097 0.6527 0.0378);"></div><span class="value">oklch(62% 0.26 145);</span></div>
  <span>}</span>
<span>}</span>
</code>
</pre>

#### 4. Взаимопонимание между разработчиками и дизайнерами

> Формат OKLCH больше всех приближен к цветам, которые мы видим в реальности. Поэтому использование полифила `oklch()` улучшило наше понимание, как можно работать с цветом.

Reviewer: ```suggestion
Математика OKLCH ближе к тому, как работают наши глаза. Поэтому, работая с `oklch()`, мы глубже понимаем как работает цвет. 
```

> Кроме того, с OKLCH дизайнеры и разработчики стали лучше друг друга понимать. Мы можем использовать `oklch()` как в инструментах для дизайна (к примеру, в [генераторах палитр](https://huetone.ardov.me/)), так и для разработки. Это помогает добиться одинакового видения сайта с обеих сторон.

Reviewer: ```suggestion
А понимание работы цвета, позволяет разработчикам и дизайнерам стать ближе друг к другу. Современные [генераторы палитр](https://huetone.ardov.me/) уже используют Oklab или Lab для хорошей доступности — использование `oklch()` в коде помогает добиться одинакового видения сайта с обеих сторон.
```

Reviewer: Figma-то не поддерживает OKLCH, поэтому нельзя говорить `как в инструментах для дизайна`

#### 5. Готовность к будущему

Перейдя на OKLCH сегодня, мы подготовили к себя к завтрашнему дню, когда в CSS появится встроенная работа с цветом.
OKLCH — лучшее пространство для работы с цветом, и мы советуем познакомиться с ним как можно скорее.
<pre data-lang="css">
<code tabindex="0" class="language-css">
<span class="selector">.button:hover&nbsp;</span><span>{</span>
  <span class="property">background:&nbsp;</span><div class="preview-with-value"><span class="value">oklch(from var(--button-color) calc(l + 10%) c h);</span></div>
<span>}</span>
</code>

> </pre>

Reviewer: Да, окончание кажется неподходящим для сайта Веб-стандартов. Давай его опустим.

Author: А тебе не показался конец статьи в текущем виде оборванным? Прерывается на куске кода.

Reviewer: Может что-то в стиле «И вы попробуйте добавить `oklch()` в ваш проект. Если у вас будут вопросы, пишите [наши твиттеры].»?

Author: Да, тоже вижу это примерно так. Попробовала сформулировать следующим образом:

```
Попробуйте и вы добавить `oklch()` в свой проект. Мы уверены, что результат вам понравится.
Если у вас возникнут вопросы, пишите нам, [@andrey_sitnik](ссылка в твиттер) и [@ninoid_](ссылка в твиттер).
```
